Allowlist character classes for post titles

jgaskins opened the pull request:
<!--
     For Work In Progress Pull Requests, please use the Draft PR feature,
     see https://github.blog/2019-02-14-introducing-draft-pull-requests/ for further details.

     For a timely review/response, please avoid force-pushing additional
     commits if your PR already received reviews or comments.

     Before submitting a Pull Request, please ensure you've done the following:
     - 📖 Read the Forem Contributing Guide: https://github.com/forem/forem/blob/main/CONTRIBUTING.md#create-a-pull-request.
     - 📖 Read the Forem Code of Conduct: https://github.com/forem/forem/blob/main/CODE_OF_CONDUCT.md.
     - 👷‍♀️ Create small PRs. In most cases this will be possible.
     - ✅ Provide tests for your changes.
     - 📝 Use descriptive commit messages.
     - 📗 Update any related documentation and include any relevant screenshots.

     NOTE: Pull Requests from forked repositories will need to be reviewed by
     a Forem Team member before any CI builds will run. Once your PR is approved
     with a `/ci` reply to the PR, it will be allowed to run subsequent builds without
     manual approval.
-->

## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Rather than deny-listing character classes we want removed, we can instead allow-list the safe ones and remove everything else.

This is in response to a post that broke the DEV homepage yesterday. Its title didn't just use whitespace, which a browser would've merged into a single space, but actually used a whitespace-looking character (`\u2800`, ["Braille Pattern Blank"](https://www.compart.com/en/unicode/U+2800)) so that that wouldn't happen and deliberately messed up the formatting not only of their own post, but any feed the post would appear as a recommended post on (including the homepage).

This PR allow-lists safe character classes:
- Word characters (letters, numbers, underscores)
- Whitespace (spaces, tabs, newlines, etc)
- Punctuation
- Emojis that I could think of

Everything else gets converted into a space character and then all whitespace gets merged into a single space, not just for rendering but for storage, as well.

I'm sure I'm not covering all the bases here, so if anyone has any more examples of characters to add into the test, I'd be happy to update it.

## Related Tickets & Documents

- https://dev.to/inhuofficial/i-am-warning-you-dont-click-42im

## QA Instructions, Screenshots, Recordings

Copy and paste the title from [this article](https://dev.to/inhuofficial/i-am-warning-you-dont-click-42im) and any others you can think of that have formatting implications.

### UI accessibility concerns?

_If your PR includes UI changes, please replace this line with details on how
accessibility is impacted and tested. For more info, check out the
[Forem Accessibility Docs](https://developers.forem.com/frontend/accessibility)._

## Added/updated tests?

- [x] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

Also, tests could be more comprehensive

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://developers.forem.com) or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] This PR changes the Forem platform and our documentation needs to be
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
- [ ] I've updated the README or added inline documentation
- [ ] I've added an entry to
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
- [x] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [ ] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

jgaskins commented on spec/models/article_spec.rb:
```diff
@@ -222,6 +222,15 @@ def build_and_validate_article(*args)
+      it "removes extra spaces" do
+        # Rubocop objects to this for many reasons, but it's intentional
+        # rubocop:disable
+        article = create(:article, title: "I⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀Am⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀Warning⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀You⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀Don't⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀Click!")
```
1. Is `rubocop:disable` not the right directive here?
2. The formatting of the suggested diff seems worse than the long line
3. Why do we even have a linter running against test code? Surely we have to do things in tests that would be terrible for production code.

jeremyf replied:
The linters are somewhat different (like don't allow over nesting of specs, allow large blocks)

jgaskins replied:
Ahh, TIL. Most projects I've worked on, the test suite wasn't linted. Usually if it was, it was an oversight. I didn't realize we were doing it deliberately.

jeremyf commented on spec/models/article_spec.rb:
```diff
@@ -222,6 +222,15 @@ def build_and_validate_article(*args)
     describe "title validation" do
+      it "removes extra spaces" do
+        # Rubocop objects to this for many reasons, but it's intentional
+        # rubocop:disable
```
`rubocop:disable Layout/LineLength` perhaps?

jgaskins replied:
It had a couple different objections to this line so I just tried to disable all of them. It was hard to tell locally, though, because Rubocop was examining this entire file (not just the diff) and it objected to lots of things in the file. I assume new rules were added with Rubocop updates. But it was just a mess with stuff that had nothing to do with my PR, so I skipped the local linting entirely. 😕

jeremyf commented on app/models/article.rb:
```diff
@@ -567,6 +568,17 @@ def set_caches
+  def normalize_title
+    return unless title
+
+    self.title = title
```
Wondering about leveraging a new method for https://github.com/forem/forem/blob/fbf16f54cef0d47ad07b35d52d6f18dc2c88e4f0/app/lib/string_attribute_cleaner.rb#L1

Like we have nullify but we don't have normalize.  (that could be a second refactor)

jgaskins replied:
@jeremyf This is interesting. I wonder if we'd need to use this for things other than post titles. Maybe display names for users, since author names are shown next to post titles in the list?

jeremyf replied:
I'm starting working on it (because my "regular" work's done for the week and I'm waiting on a sleeping puppy)

aitchiss reviewed (approved):
Nice! I think this is a reasonable action to take for post titles in particular 👍

djuber commented on app/models/article.rb:
```diff
@@ -567,6 +568,17 @@ def set_caches
+      # recognized emoji
+      .gsub(/[^[:word:][:space:][:punct:]\u{1f000}-\u{1ffff}]+/, " ")
+      # Coalesce runs of whitespace into a single space character
+      .gsub(/\s+/, " ")
```
not sure if a final `strip` or `trim` makes sense (remove space on the ends), but a single space character seems generally harmless.

djuber commented on app/models/article.rb:
```diff
@@ -567,6 +568,17 @@ def set_caches
+    self.title = title
+      # Filter out anything that isn't a word, space, punctuation mark, or
+      # recognized emoji
+      .gsub(/[^[:word:][:space:][:punct:]\u{1f000}-\u{1ffff}]+/, " ")
```
If, and when, we need to reuse this regex pattern, it probably should have a name (either as a constant or an abstracted method).

This pattern of allowed characters sounds like it's the least certain part of the PR, and most likely to be required to change in the future, and may belong someplace outside of article (single source of truth, one reason to change).

djuber reviewed (approved):
Do we have other things with rendered titles that could cause layout issues, for example could a wonky podcast episode cause similar issue?

aitchiss commented:
> Do we have other things with rendered titles that could cause layout issues, for example could a wonky podcast episode cause similar issue?

Yeah definitely this can happen in other areas, but I guess we need to decide how much we address via the allowlist, and how much we address in the UI. I have a PR that addresses the same from the UI side of things - https://github.com/forem/forem/pull/16782

jgaskins commented:
> we need to decide how much we address via the allowlist, and how much we address in the UI.

@aitchiss This is a really good point. Obviously we want the UI to look okay no matter what content is being rendered, so your PR makes perfect sense regardless of what we do on the back end. This PR is currently limited to article titles, so there may be other things that your fix will catch that the backend currently does not, but should. Robustness of the application as a whole tends to come from multiple layers being robust rather than each layer only handling disjoint subsets of the whole.

jgaskins commented:
Regarding the range of accepted emoji, the range put in the regex covers a lot of it but is incomplete. Looks like there are some emoji outside the `1f000-1ffff` range, according to [this StackOverflow post](https://stackoverflow.com/questions/30470079/emoji-value-range).

I wrote a quick Ruby script to show the emoji from those ranges:

```ruby
codepoints = "U+00A9, U+00AE, U+203C, U+2049, U+20E3, U+2122, U+2139, U+2194-2199, U+21A9-21AA, U+231A, U+231B, U+2328, U+23CF, U+23E9-23F3, U+23F8-23FA, U+24C2, U+25AA, U+25AB, U+25B6, U+25C0, U+25FB-25FE, U+2600-27EF, U+2934, U+2935, U+2B00-2BFF, U+3030, U+303D, U+3297, U+3299, U+1F000-1F02F, U+1F0A0-1F0FF, U+1F100-1F64F, U+1F680-1F6FF, U+1F910-1F96B, U+1F980-1F9E0"
  .split(', ')
  .flat_map do |codepoint_or_range|
    codepoint_or_range = codepoint_or_range.match(/U\+(.*)/)[1]
    if codepoint_or_range.include? '-'
      low, high = codepoint_or_range.split('-')
      (low.to_i(16)..high.to_i(16)).to_a
    else
      codepoint_or_range.to_i(16)
    end
  end

pp codepoints.map { |cp| [cp.to_s(16), cp.chr(Encoding::UTF_8)] }.to_h
```

The output is collapsed below:

<details><summary>Mapping of codepoints to emoji</summary>

{"a9"=>"©",
 "ae"=>"®",
 "203c"=>"‼",
 "2049"=>"⁉",
 "20e3"=>"⃣",
 "2122"=>"™",
 "2139"=>"ℹ",
 "2194"=>"↔",
 "2195"=>"↕",
 "2196"=>"↖",
 "2197"=>"↗",
 "2198"=>"↘",
 "2199"=>"↙",
 "21a9"=>"↩",
 "21aa"=>"↪",
 "231a"=>"⌚",
 "231b"=>"⌛",
 "2328"=>"⌨",
 "23cf"=>"⏏",
 "23e9"=>"⏩",
 "23ea"=>"⏪",
 "23eb"=>"⏫",
 "23ec"=>"⏬",
 "23ed"=>"⏭",
 "23ee"=>"⏮",
 "23ef"=>"⏯",
 "23f0"=>"⏰",
 "23f1"=>"⏱",
 "23f2"=>"⏲",
 "23f3"=>"⏳",
 "23f8"=>"⏸",
 "23f9"=>"⏹",
 "23fa"=>"⏺",
 "24c2"=>"Ⓜ",
 "25aa"=>"▪",
 "25ab"=>"▫",
 "25b6"=>"▶",
 "25c0"=>"◀",
 "25fb"=>"◻",
 "25fc"=>"◼",
 "25fd"=>"◽",
 "25fe"=>"◾",
 "2600"=>"☀",
 "2601"=>"☁",
 "2602"=>"☂",
 "2603"=>"☃",
 "2604"=>"☄",
 "2605"=>"★",
 "2606"=>"☆",
 "2607"=>"☇",
 "2608"=>"☈",
 "2609"=>"☉",
 "260a"=>"☊",
 "260b"=>"☋",
 "260c"=>"☌",
 "260d"=>"☍",
 "260e"=>"☎",
 "260f"=>"☏",
 "2610"=>"☐",
 "2611"=>"☑",
 "2612"=>"☒",
 "2613"=>"☓",
 "2614"=>"☔",
 "2615"=>"☕",
 "2616"=>"☖",
 "2617"=>"☗",
 "2618"=>"☘",
 "2619"=>"☙",
 "261a"=>"☚",
 "261b"=>"☛",
 "261c"=>"☜",
 "261d"=>"☝",
 "261e"=>"☞",
 "261f"=>"☟",
 "2620"=>"☠",
 "2621"=>"☡",
 "2622"=>"☢",
 "2623"=>"☣",
 "2624"=>"☤",
 "2625"=>"☥",
 "2626"=>"☦",
 "2627"=>"☧",
 "2628"=>"☨",
 "2629"=>"☩",
 "262a"=>"☪",
 "262b"=>"☫",
 "262c"=>"☬",
 "262d"=>"☭",
 "262e"=>"☮",
 "262f"=>"☯",
 "2630"=>"☰",
 "2631"=>"☱",
 "2632"=>"☲",
 "2633"=>"☳",
 "2634"=>"☴",
 "2635"=>"☵",
 "2636"=>"☶",
 "2637"=>"☷",
 "2638"=>"☸",
 "2639"=>"☹",
 "263a"=>"☺",
 "263b"=>"☻",
 "263c"=>"☼",
 "263d"=>"☽",
 "263e"=>"☾",
 "263f"=>"☿",
 "2640"=>"♀",
 "2641"=>"♁",
 "2642"=>"♂",
 "2643"=>"♃",
 "2644"=>"♄",
 "2645"=>"♅",
 "2646"=>"♆",
 "2647"=>"♇",
 "2648"=>"♈",
 "2649"=>"♉",
 "264a"=>"♊",
 "264b"=>"♋",
 "264c"=>"♌",
 "264d"=>"♍",
 "264e"=>"♎",
 "264f"=>"♏",
 "2650"=>"♐",
 "2651"=>"♑",
 "2652"=>"♒",
 "2653"=>"♓",
 "2654"=>"♔",
 "2655"=>"♕",
 "2656"=>"♖",
 "2657"=>"♗",
 "2658"=>"♘",
 "2659"=>"♙",
 "265a"=>"♚",
 "265b"=>"♛",
 "265c"=>"♜",
 "265d"=>"♝",
 "265e"=>"♞",
 "265f"=>"♟",
 "2660"=>"♠",
 "2661"=>"♡",
 "2662"=>"♢",
 "2663"=>"♣",
 "2664"=>"♤",
 "2665"=>"♥",
 "2666"=>"♦",
 "2667"=>"♧",
 "2668"=>"♨",
 "2669"=>"♩",
 "266a"=>"♪",
 "266b"=>"♫",
 "266c"=>"♬",
 "266d"=>"♭",
 "266e"=>"♮",
 "266f"=>"♯",
 "2670"=>"♰",
 "2671"=>"♱",
 "2672"=>"♲",
 "2673"=>"♳",
 "2674"=>"♴",
 "2675"=>"♵",
 "2676"=>"♶",
 "2677"=>"♷",
 "2678"=>"♸",
 "2679"=>"♹",
 "267a"=>"♺",
 "267b"=>"♻",
 "267c"=>"♼",
 "267d"=>"♽",
 "267e"=>"♾",
 "267f"=>"♿",
 "2680"=>"⚀",
 "2681"=>"⚁",
 "2682"=>"⚂",
 "2683"=>"⚃",
 "2684"=>"⚄",
 "2685"=>"⚅",
 "2686"=>"⚆",
 "2687"=>"⚇",
 "2688"=>"⚈",
 "2689"=>"⚉",
 "268a"=>"⚊",
 "268b"=>"⚋",
 "268c"=>"⚌",
 "268d"=>"⚍",
 "268e"=>"⚎",
 "268f"=>"⚏",
 "2690"=>"⚐",
 "2691"=>"⚑",
 "2692"=>"⚒",
 "2693"=>"⚓",
 "2694"=>"⚔",
 "2695"=>"⚕",
 "2696"=>"⚖",
 "2697"=>"⚗",
 "2698"=>"⚘",
 "2699"=>"⚙",
 "269a"=>"⚚",
 "269b"=>"⚛",
 "269c"=>"⚜",
 "269d"=>"⚝",
 "269e"=>"⚞",
 "269f"=>"⚟",
 "26a0"=>"⚠",
 "26a1"=>"⚡",
 "26a2"=>"⚢",
 "26a3"=>"⚣",
 "26a4"=>"⚤",
 "26a5"=>"⚥",
 "26a6"=>"⚦",
 "26a7"=>"⚧",
 "26a8"=>"⚨",
 "26a9"=>"⚩",
 "26aa"=>"⚪",
 "26ab"=>"⚫",
 "26ac"=>"⚬",
 "26ad"=>"⚭",
 "26ae"=>"⚮",
 "26af"=>"⚯",
 "26b0"=>"⚰",
 "26b1"=>"⚱",
 "26b2"=>"⚲",
 "26b3"=>"⚳",
 "26b4"=>"⚴",
 "26b5"=>"⚵",
 "26b6"=>"⚶",
 "26b7"=>"⚷",
 "26b8"=>"⚸",
 "26b9"=>"⚹",
 "26ba"=>"⚺",
 "26bb"=>"⚻",
 "26bc"=>"⚼",
 "26bd"=>"⚽",
 "26be"=>"⚾",
 "26bf"=>"⚿",
 "26c0"=>"⛀",
 "26c1"=>"⛁",
 "26c2"=>"⛂",
 "26c3"=>"⛃",
 "26c4"=>"⛄",
 "26c5"=>"⛅",
 "26c6"=>"⛆",
 "26c7"=>"⛇",
 "26c8"=>"⛈",
 "26c9"=>"⛉",
 "26ca"=>"⛊",
 "26cb"=>"⛋",
 "26cc"=>"⛌",
 "26cd"=>"⛍",
 "26ce"=>"⛎",
 "26cf"=>"⛏",
 "26d0"=>"⛐",
 "26d1"=>"⛑",
 "26d2"=>"⛒",
 "26d3"=>"⛓",
 "26d4"=>"⛔",
 "26d5"=>"⛕",
 "26d6"=>"⛖",
 "26d7"=>"⛗",
 "26d8"=>"⛘",
 "26d9"=>"⛙",
 "26da"=>"⛚",
 "26db"=>"⛛",
 "26dc"=>"⛜",
 "26dd"=>"⛝",
 "26de"=>"⛞",
 "26df"=>"⛟",
 "26e0"=>"⛠",
 "26e1"=>"⛡",
 "26e2"=>"⛢",
 "26e3"=>"⛣",
 "26e4"=>"⛤",
 "26e5"=>"⛥",
 "26e6"=>"⛦",
 "26e7"=>"⛧",
 "26e8"=>"⛨",
 "26e9"=>"⛩",
 "26ea"=>"⛪",
 "26eb"=>"⛫",
 "26ec"=>"⛬",
 "26ed"=>"⛭",
 "26ee"=>"⛮",
 "26ef"=>"⛯",
 "26f0"=>"⛰",
 "26f1"=>"⛱",
 "26f2"=>"⛲",
 "26f3"=>"⛳",
 "26f4"=>"⛴",
 "26f5"=>"⛵",
 "26f6"=>"⛶",
 "26f7"=>"⛷",
 "26f8"=>"⛸",
 "26f9"=>"⛹",
 "26fa"=>"⛺",
 "26fb"=>"⛻",
 "26fc"=>"⛼",
 "26fd"=>"⛽",
 "26fe"=>"⛾",
 "26ff"=>"⛿",
 "2700"=>"✀",
 "2701"=>"✁",
 "2702"=>"✂",
 "2703"=>"✃",
 "2704"=>"✄",
 "2705"=>"✅",
 "2706"=>"✆",
 "2707"=>"✇",
 "2708"=>"✈",
 "2709"=>"✉",
 "270a"=>"✊",
 "270b"=>"✋",
 "270c"=>"✌",
 "270d"=>"✍",
 "270e"=>"✎",
 "270f"=>"✏",
 "2710"=>"✐",
 "2711"=>"✑",
 "2712"=>"✒",
 "2713"=>"✓",
 "2714"=>"✔",
 "2715"=>"✕",
 "2716"=>"✖",
 "2717"=>"✗",
 "2718"=>"✘",
 "2719"=>"✙",
 "271a"=>"✚",
 "271b"=>"✛",
 "271c"=>"✜",
 "271d"=>"✝",
 "271e"=>"✞",
 "271f"=>"✟",
 "2720"=>"✠",
 "2721"=>"✡",
 "2722"=>"✢",
 "2723"=>"✣",
 "2724"=>"✤",
 "2725"=>"✥",
 "2726"=>"✦",
 "2727"=>"✧",
 "2728"=>"✨",
 "2729"=>"✩",
 "272a"=>"✪",
 "272b"=>"✫",
 "272c"=>"✬",
 "272d"=>"✭",
 "272e"=>"✮",
 "272f"=>"✯",
 "2730"=>"✰",
 "2731"=>"✱",
 "2732"=>"✲",
 "2733"=>"✳",
 "2734"=>"✴",
 "2735"=>"✵",
 "2736"=>"✶",
 "2737"=>"✷",
 "2738"=>"✸",
 "2739"=>"✹",
 "273a"=>"✺",
 "273b"=>"✻",
 "273c"=>"✼",
 "273d"=>"✽",
 "273e"=>"✾",
 "273f"=>"✿",
 "2740"=>"❀",
 "2741"=>"❁",
 "2742"=>"❂",
 "2743"=>"❃",
 "2744"=>"❄",
 "2745"=>"❅",
 "2746"=>"❆",
 "2747"=>"❇",
 "2748"=>"❈",
 "2749"=>"❉",
 "274a"=>"❊",
 "274b"=>"❋",
 "274c"=>"❌",
 "274d"=>"❍",
 "274e"=>"❎",
 "274f"=>"❏",
 "2750"=>"❐",
 "2751"=>"❑",
 "2752"=>"❒",
 "2753"=>"❓",
 "2754"=>"❔",
 "2755"=>"❕",
 "2756"=>"❖",
 "2757"=>"❗",
 "2758"=>"❘",
 "2759"=>"❙",
 "275a"=>"❚",
 "275b"=>"❛",
 "275c"=>"❜",
 "275d"=>"❝",
 "275e"=>"❞",
 "275f"=>"❟",
 "2760"=>"❠",
 "2761"=>"❡",
 "2762"=>"❢",
 "2763"=>"❣",
 "2764"=>"❤",
 "2765"=>"❥",
 "2766"=>"❦",
 "2767"=>"❧",
 "2768"=>"❨",
 "2769"=>"❩",
 "276a"=>"❪",
 "276b"=>"❫",
 "276c"=>"❬",
 "276d"=>"❭",
 "276e"=>"❮",
 "276f"=>"❯",
 "2770"=>"❰",
 "2771"=>"❱",
 "2772"=>"❲",
 "2773"=>"❳",
 "2774"=>"❴",
 "2775"=>"❵",
 "2776"=>"❶",
 "2777"=>"❷",
 "2778"=>"❸",
 "2779"=>"❹",
 "277a"=>"❺",
 "277b"=>"❻",
 "277c"=>"❼",
 "277d"=>"❽",
 "277e"=>"❾",
 "277f"=>"❿",
 "2780"=>"➀",
 "2781"=>"➁",
 "2782"=>"➂",
 "2783"=>"➃",
 "2784"=>"➄",
 "2785"=>"➅",
 "2786"=>"➆",
 "2787"=>"➇",
 "2788"=>"➈",
 "2789"=>"➉",
 "278a"=>"➊",
 "278b"=>"➋",
 "278c"=>"➌",
 "278d"=>"➍",
 "278e"=>"➎",
 "278f"=>"➏",
 "2790"=>"➐",
 "2791"=>"➑",
 "2792"=>"➒",
 "2793"=>"➓",
 "2794"=>"➔",
 "2795"=>"➕",
 "2796"=>"➖",
 "2797"=>"➗",
 "2798"=>"➘",
 "2799"=>"➙",
 "279a"=>"➚",
 "279b"=>"➛",
 "279c"=>"➜",
 "279d"=>"➝",
 "279e"=>"➞",
 "279f"=>"➟",
 "27a0"=>"➠",
 "27a1"=>"➡",
 "27a2"=>"➢",
 "27a3"=>"➣",
 "27a4"=>"➤",
 "27a5"=>"➥",
 "27a6"=>"➦",
 "27a7"=>"➧",
 "27a8"=>"➨",
 "27a9"=>"➩",
 "27aa"=>"➪",
 "27ab"=>"➫",
 "27ac"=>"➬",
 "27ad"=>"➭",
 "27ae"=>"➮",
 "27af"=>"➯",
 "27b0"=>"➰",
 "27b1"=>"➱",
 "27b2"=>"➲",
 "27b3"=>"➳",
 "27b4"=>"➴",
 "27b5"=>"➵",
 "27b6"=>"➶",
 "27b7"=>"➷",
 "27b8"=>"➸",
 "27b9"=>"➹",
 "27ba"=>"➺",
 "27bb"=>"➻",
 "27bc"=>"➼",
 "27bd"=>"➽",
 "27be"=>"➾",
 "27bf"=>"➿",
 "27c0"=>"⟀",
 "27c1"=>"⟁",
 "27c2"=>"⟂",
 "27c3"=>"⟃",
 "27c4"=>"⟄",
 "27c5"=>"⟅",
 "27c6"=>"⟆",
 "27c7"=>"⟇",
 "27c8"=>"⟈",
 "27c9"=>"⟉",
 "27ca"=>"⟊",
 "27cb"=>"⟋",
 "27cc"=>"⟌",
 "27cd"=>"⟍",
 "27ce"=>"⟎",
 "27cf"=>"⟏",
 "27d0"=>"⟐",
 "27d1"=>"⟑",
 "27d2"=>"⟒",
 "27d3"=>"⟓",
 "27d4"=>"⟔",
 "27d5"=>"⟕",
 "27d6"=>"⟖",
 "27d7"=>"⟗",
 "27d8"=>"⟘",
 "27d9"=>"⟙",
 "27da"=>"⟚",
 "27db"=>"⟛",
 "27dc"=>"⟜",
 "27dd"=>"⟝",
 "27de"=>"⟞",
 "27df"=>"⟟",
 "27e0"=>"⟠",
 "27e1"=>"⟡",
 "27e2"=>"⟢",
 "27e3"=>"⟣",
 "27e4"=>"⟤",
 "27e5"=>"⟥",
 "27e6"=>"⟦",
 "27e7"=>"⟧",
 "27e8"=>"⟨",
 "27e9"=>"⟩",
 "27ea"=>"⟪",
 "27eb"=>"⟫",
 "27ec"=>"⟬",
 "27ed"=>"⟭",
 "27ee"=>"⟮",
 "27ef"=>"⟯",
 "2934"=>"⤴",
 "2935"=>"⤵",
 "2b00"=>"⬀",
 "2b01"=>"⬁",
 "2b02"=>"⬂",
 "2b03"=>"⬃",
 "2b04"=>"⬄",
 "2b05"=>"⬅",
 "2b06"=>"⬆",
 "2b07"=>"⬇",
 "2b08"=>"⬈",
 "2b09"=>"⬉",
 "2b0a"=>"⬊",
 "2b0b"=>"⬋",
 "2b0c"=>"⬌",
 "2b0d"=>"⬍",
 "2b0e"=>"⬎",
 "2b0f"=>"⬏",
 "2b10"=>"⬐",
 "2b11"=>"⬑",
 "2b12"=>"⬒",
 "2b13"=>"⬓",
 "2b14"=>"⬔",
 "2b15"=>"⬕",
 "2b16"=>"⬖",
 "2b17"=>"⬗",
 "2b18"=>"⬘",
 "2b19"=>"⬙",
 "2b1a"=>"⬚",
 "2b1b"=>"⬛",
 "2b1c"=>"⬜",
 "2b1d"=>"⬝",
 "2b1e"=>"⬞",
 "2b1f"=>"⬟",
 "2b20"=>"⬠",
 "2b21"=>"⬡",
 "2b22"=>"⬢",
 "2b23"=>"⬣",
 "2b24"=>"⬤",
 "2b25"=>"⬥",
 "2b26"=>"⬦",
 "2b27"=>"⬧",
 "2b28"=>"⬨",
 "2b29"=>"⬩",
 "2b2a"=>"⬪",
 "2b2b"=>"⬫",
 "2b2c"=>"⬬",
 "2b2d"=>"⬭",
 "2b2e"=>"⬮",
 "2b2f"=>"⬯",
 "2b30"=>"⬰",
 "2b31"=>"⬱",
 "2b32"=>"⬲",
 "2b33"=>"⬳",
 "2b34"=>"⬴",
 "2b35"=>"⬵",
 "2b36"=>"⬶",
 "2b37"=>"⬷",
 "2b38"=>"⬸",
 "2b39"=>"⬹",
 "2b3a"=>"⬺",
 "2b3b"=>"⬻",
 "2b3c"=>"⬼",
 "2b3d"=>"⬽",
 "2b3e"=>"⬾",
 "2b3f"=>"⬿",
 "2b40"=>"⭀",
 "2b41"=>"⭁",
 "2b42"=>"⭂",
 "2b43"=>"⭃",
 "2b44"=>"⭄",
 "2b45"=>"⭅",
 "2b46"=>"⭆",
 "2b47"=>"⭇",
 "2b48"=>"⭈",
 "2b49"=>"⭉",
 "2b4a"=>"⭊",
 "2b4b"=>"⭋",
 "2b4c"=>"⭌",
 "2b4d"=>"⭍",
 "2b4e"=>"⭎",
 "2b4f"=>"⭏",
 "2b50"=>"⭐",
 "2b51"=>"⭑",
 "2b52"=>"⭒",
 "2b53"=>"⭓",
 "2b54"=>"⭔",
 "2b55"=>"⭕",
 "2b56"=>"⭖",
 "2b57"=>"⭗",
 "2b58"=>"⭘",
 "2b59"=>"⭙",
 "2b5a"=>"⭚",
 "2b5b"=>"⭛",
 "2b5c"=>"⭜",
 "2b5d"=>"⭝",
 "2b5e"=>"⭞",
 "2b5f"=>"⭟",
 "2b60"=>"⭠",
 "2b61"=>"⭡",
 "2b62"=>"⭢",
 "2b63"=>"⭣",
 "2b64"=>"⭤",
 "2b65"=>"⭥",
 "2b66"=>"⭦",
 "2b67"=>"⭧",
 "2b68"=>"⭨",
 "2b69"=>"⭩",
 "2b6a"=>"⭪",
 "2b6b"=>"⭫",
 "2b6c"=>"⭬",
 "2b6d"=>"⭭",
 "2b6e"=>"⭮",
 "2b6f"=>"⭯",
 "2b70"=>"⭰",
 "2b71"=>"⭱",
 "2b72"=>"⭲",
 "2b73"=>"⭳",
 "2b74"=>"\u2B74",
 "2b75"=>"\u2B75",
 "2b76"=>"⭶",
 "2b77"=>"⭷",
 "2b78"=>"⭸",
 "2b79"=>"⭹",
 "2b7a"=>"⭺",
 "2b7b"=>"⭻",
 "2b7c"=>"⭼",
 "2b7d"=>"⭽",
 "2b7e"=>"⭾",
 "2b7f"=>"⭿",
 "2b80"=>"⮀",
 "2b81"=>"⮁",
 "2b82"=>"⮂",
 "2b83"=>"⮃",
 "2b84"=>"⮄",
 "2b85"=>"⮅",
 "2b86"=>"⮆",
 "2b87"=>"⮇",
 "2b88"=>"⮈",
 "2b89"=>"⮉",
 "2b8a"=>"⮊",
 "2b8b"=>"⮋",
 "2b8c"=>"⮌",
 "2b8d"=>"⮍",
 "2b8e"=>"⮎",
 "2b8f"=>"⮏",
 "2b90"=>"⮐",
 "2b91"=>"⮑",
 "2b92"=>"⮒",
 "2b93"=>"⮓",
 "2b94"=>"⮔",
 "2b95"=>"⮕",
 "2b96"=>"\u2B96",
 "2b97"=>"⮗",
 "2b98"=>"⮘",
 "2b99"=>"⮙",
 "2b9a"=>"⮚",
 "2b9b"=>"⮛",
 "2b9c"=>"⮜",
 "2b9d"=>"⮝",
 "2b9e"=>"⮞",
 "2b9f"=>"⮟",
 "2ba0"=>"⮠",
 "2ba1"=>"⮡",
 "2ba2"=>"⮢",
 "2ba3"=>"⮣",
 "2ba4"=>"⮤",
 "2ba5"=>"⮥",
 "2ba6"=>"⮦",
 "2ba7"=>"⮧",
 "2ba8"=>"⮨",
 "2ba9"=>"⮩",
 "2baa"=>"⮪",
 "2bab"=>"⮫",
 "2bac"=>"⮬",
 "2bad"=>"⮭",
 "2bae"=>"⮮",
 "2baf"=>"⮯",
 "2bb0"=>"⮰",
 "2bb1"=>"⮱",
 "2bb2"=>"⮲",
 "2bb3"=>"⮳",
 "2bb4"=>"⮴",
 "2bb5"=>"⮵",
 "2bb6"=>"⮶",
 "2bb7"=>"⮷",
 "2bb8"=>"⮸",
 "2bb9"=>"⮹",
 "2bba"=>"⮺",
 "2bbb"=>"⮻",
 "2bbc"=>"⮼",
 "2bbd"=>"⮽",
 "2bbe"=>"⮾",
 "2bbf"=>"⮿",
 "2bc0"=>"⯀",
 "2bc1"=>"⯁",
 "2bc2"=>"⯂",
 "2bc3"=>"⯃",
 "2bc4"=>"⯄",
 "2bc5"=>"⯅",
 "2bc6"=>"⯆",
 "2bc7"=>"⯇",
 "2bc8"=>"⯈",
 "2bc9"=>"⯉",
 "2bca"=>"⯊",
 "2bcb"=>"⯋",
 "2bcc"=>"⯌",
 "2bcd"=>"⯍",
 "2bce"=>"⯎",
 "2bcf"=>"⯏",
 "2bd0"=>"⯐",
 "2bd1"=>"⯑",
 "2bd2"=>"⯒",
 "2bd3"=>"⯓",
 "2bd4"=>"⯔",
 "2bd5"=>"⯕",
 "2bd6"=>"⯖",
 "2bd7"=>"⯗",
 "2bd8"=>"⯘",
 "2bd9"=>"⯙",
 "2bda"=>"⯚",
 "2bdb"=>"⯛",
 "2bdc"=>"⯜",
 "2bdd"=>"⯝",
 "2bde"=>"⯞",
 "2bdf"=>"⯟",
 "2be0"=>"⯠",
 "2be1"=>"⯡",
 "2be2"=>"⯢",
 "2be3"=>"⯣",
 "2be4"=>"⯤",
 "2be5"=>"⯥",
 "2be6"=>"⯦",
 "2be7"=>"⯧",
 "2be8"=>"⯨",
 "2be9"=>"⯩",
 "2bea"=>"⯪",
 "2beb"=>"⯫",
 "2bec"=>"⯬",
 "2bed"=>"⯭",
 "2bee"=>"⯮",
 "2bef"=>"⯯",
 "2bf0"=>"⯰",
 "2bf1"=>"⯱",
 "2bf2"=>"⯲",
 "2bf3"=>"⯳",
 "2bf4"=>"⯴",
 "2bf5"=>"⯵",
 "2bf6"=>"⯶",
 "2bf7"=>"⯷",
 "2bf8"=>"⯸",
 "2bf9"=>"⯹",
 "2bfa"=>"⯺",
 "2bfb"=>"⯻",
 "2bfc"=>"⯼",
 "2bfd"=>"⯽",
 "2bfe"=>"⯾",
 "2bff"=>"⯿",
 "3030"=>"〰",
 "303d"=>"〽",
 "3297"=>"㊗",
 "3299"=>"㊙",
 "1f000"=>"🀀",
 "1f001"=>"🀁",
 "1f002"=>"🀂",
 "1f003"=>"🀃",
 "1f004"=>"🀄",
 "1f005"=>"🀅",
 "1f006"=>"🀆",
 "1f007"=>"🀇",
 "1f008"=>"🀈",
 "1f009"=>"🀉",
 "1f00a"=>"🀊",
 "1f00b"=>"🀋",
 "1f00c"=>"🀌",
 "1f00d"=>"🀍",
 "1f00e"=>"🀎",
 "1f00f"=>"🀏",
 "1f010"=>"🀐",
 "1f011"=>"🀑",
 "1f012"=>"🀒",
 "1f013"=>"🀓",
 "1f014"=>"🀔",
 "1f015"=>"🀕",
 "1f016"=>"🀖",
 "1f017"=>"🀗",
 "1f018"=>"🀘",
 "1f019"=>"🀙",
 "1f01a"=>"🀚",
 "1f01b"=>"🀛",
 "1f01c"=>"🀜",
 "1f01d"=>"🀝",
 "1f01e"=>"🀞",
 "1f01f"=>"🀟",
 "1f020"=>"🀠",
 "1f021"=>"🀡",
 "1f022"=>"🀢",
 "1f023"=>"🀣",
 "1f024"=>"🀤",
 "1f025"=>"🀥",
 "1f026"=>"🀦",
 "1f027"=>"🀧",
 "1f028"=>"🀨",
 "1f029"=>"🀩",
 "1f02a"=>"🀪",
 "1f02b"=>"🀫",
 "1f02c"=>"\u{1F02C}",
 "1f02d"=>"\u{1F02D}",
 "1f02e"=>"\u{1F02E}",
 "1f02f"=>"\u{1F02F}",
 "1f0a0"=>"🂠",
 "1f0a1"=>"🂡",
 "1f0a2"=>"🂢",
 "1f0a3"=>"🂣",
 "1f0a4"=>"🂤",
 "1f0a5"=>"🂥",
 "1f0a6"=>"🂦",
 "1f0a7"=>"🂧",
 "1f0a8"=>"🂨",
 "1f0a9"=>"🂩",
 "1f0aa"=>"🂪",
 "1f0ab"=>"🂫",
 "1f0ac"=>"🂬",
 "1f0ad"=>"🂭",
 "1f0ae"=>"🂮",
 "1f0af"=>"\u{1F0AF}",
 "1f0b0"=>"\u{1F0B0}",
 "1f0b1"=>"🂱",
 "1f0b2"=>"🂲",
 "1f0b3"=>"🂳",
 "1f0b4"=>"🂴",
 "1f0b5"=>"🂵",
 "1f0b6"=>"🂶",
 "1f0b7"=>"🂷",
 "1f0b8"=>"🂸",
 "1f0b9"=>"🂹",
 "1f0ba"=>"🂺",
 "1f0bb"=>"🂻",
 "1f0bc"=>"🂼",
 "1f0bd"=>"🂽",
 "1f0be"=>"🂾",
 "1f0bf"=>"🂿",
 "1f0c0"=>"\u{1F0C0}",
 "1f0c1"=>"🃁",
 "1f0c2"=>"🃂",
 "1f0c3"=>"🃃",
 "1f0c4"=>"🃄",
 "1f0c5"=>"🃅",
 "1f0c6"=>"🃆",
 "1f0c7"=>"🃇",
 "1f0c8"=>"🃈",
 "1f0c9"=>"🃉",
 "1f0ca"=>"🃊",
 "1f0cb"=>"🃋",
 "1f0cc"=>"🃌",
 "1f0cd"=>"🃍",
 "1f0ce"=>"🃎",
 "1f0cf"=>"🃏",
 "1f0d0"=>"\u{1F0D0}",
 "1f0d1"=>"🃑",
 "1f0d2"=>"🃒",
 "1f0d3"=>"🃓",
 "1f0d4"=>"🃔",
 "1f0d5"=>"🃕",
 "1f0d6"=>"🃖",
 "1f0d7"=>"🃗",
 "1f0d8"=>"🃘",
 "1f0d9"=>"🃙",
 "1f0da"=>"🃚",
 "1f0db"=>"🃛",
 "1f0dc"=>"🃜",
 "1f0dd"=>"🃝",
 "1f0de"=>"🃞",
 "1f0df"=>"🃟",
 "1f0e0"=>"🃠",
 "1f0e1"=>"🃡",
 "1f0e2"=>"🃢",
 "1f0e3"=>"🃣",
 "1f0e4"=>"🃤",
 "1f0e5"=>"🃥",
 "1f0e6"=>"🃦",
 "1f0e7"=>"🃧",
 "1f0e8"=>"🃨",
 "1f0e9"=>"🃩",
 "1f0ea"=>"🃪",
 "1f0eb"=>"🃫",
 "1f0ec"=>"🃬",
 "1f0ed"=>"🃭",
 "1f0ee"=>"🃮",
 "1f0ef"=>"🃯",
 "1f0f0"=>"🃰",
 "1f0f1"=>"🃱",
 "1f0f2"=>"🃲",
 "1f0f3"=>"🃳",
 "1f0f4"=>"🃴",
 "1f0f5"=>"🃵",
 "1f0f6"=>"\u{1F0F6}",
 "1f0f7"=>"\u{1F0F7}",
 "1f0f8"=>"\u{1F0F8}",
 "1f0f9"=>"\u{1F0F9}",
 "1f0fa"=>"\u{1F0FA}",
 "1f0fb"=>"\u{1F0FB}",
 "1f0fc"=>"\u{1F0FC}",
 "1f0fd"=>"\u{1F0FD}",
 "1f0fe"=>"\u{1F0FE}",
 "1f0ff"=>"\u{1F0FF}",
 "1f100"=>"🄀",
 "1f101"=>"🄁",
 "1f102"=>"🄂",
 "1f103"=>"🄃",
 "1f104"=>"🄄",
 "1f105"=>"🄅",
 "1f106"=>"🄆",
 "1f107"=>"🄇",
 "1f108"=>"🄈",
 "1f109"=>"🄉",
 "1f10a"=>"🄊",
 "1f10b"=>"🄋",
 "1f10c"=>"🄌",
 "1f10d"=>"🄍",
 "1f10e"=>"🄎",
 "1f10f"=>"🄏",
 "1f110"=>"🄐",
 "1f111"=>"🄑",
 "1f112"=>"🄒",
 "1f113"=>"🄓",
 "1f114"=>"🄔",
 "1f115"=>"🄕",
 "1f116"=>"🄖",
 "1f117"=>"🄗",
 "1f118"=>"🄘",
 "1f119"=>"🄙",
 "1f11a"=>"🄚",
 "1f11b"=>"🄛",
 "1f11c"=>"🄜",
 "1f11d"=>"🄝",
 "1f11e"=>"🄞",
 "1f11f"=>"🄟",
 "1f120"=>"🄠",
 "1f121"=>"🄡",
 "1f122"=>"🄢",
 "1f123"=>"🄣",
 "1f124"=>"🄤",
 "1f125"=>"🄥",
 "1f126"=>"🄦",
 "1f127"=>"🄧",
 "1f128"=>"🄨",
 "1f129"=>"🄩",
 "1f12a"=>"🄪",
 "1f12b"=>"🄫",
 "1f12c"=>"🄬",
 "1f12d"=>"🄭",
 "1f12e"=>"🄮",
 "1f12f"=>"🄯",
 "1f130"=>"🄰",
 "1f131"=>"🄱",
 "1f132"=>"🄲",
 "1f133"=>"🄳",
 "1f134"=>"🄴",
 "1f135"=>"🄵",
 "1f136"=>"🄶",
 "1f137"=>"🄷",
 "1f138"=>"🄸",
 "1f139"=>"🄹",
 "1f13a"=>"🄺",
 "1f13b"=>"🄻",
 "1f13c"=>"🄼",
 "1f13d"=>"🄽",
 "1f13e"=>"🄾",
 "1f13f"=>"🄿",
 "1f140"=>"🅀",
 "1f141"=>"🅁",
 "1f142"=>"🅂",
 "1f143"=>"🅃",
 "1f144"=>"🅄",
 "1f145"=>"🅅",
 "1f146"=>"🅆",
 "1f147"=>"🅇",
 "1f148"=>"🅈",
 "1f149"=>"🅉",
 "1f14a"=>"🅊",
 "1f14b"=>"🅋",
 "1f14c"=>"🅌",
 "1f14d"=>"🅍",
 "1f14e"=>"🅎",
 "1f14f"=>"🅏",
 "1f150"=>"🅐",
 "1f151"=>"🅑",
 "1f152"=>"🅒",
 "1f153"=>"🅓",
 "1f154"=>"🅔",
 "1f155"=>"🅕",
 "1f156"=>"🅖",
 "1f157"=>"🅗",
 "1f158"=>"🅘",
 "1f159"=>"🅙",
 "1f15a"=>"🅚",
 "1f15b"=>"🅛",
 "1f15c"=>"🅜",
 "1f15d"=>"🅝",
 "1f15e"=>"🅞",
 "1f15f"=>"🅟",
 "1f160"=>"🅠",
 "1f161"=>"🅡",
 "1f162"=>"🅢",
 "1f163"=>"🅣",
 "1f164"=>"🅤",
 "1f165"=>"🅥",
 "1f166"=>"🅦",
 "1f167"=>"🅧",
 "1f168"=>"🅨",
 "1f169"=>"🅩",
 "1f16a"=>"🅪",
 "1f16b"=>"🅫",
 "1f16c"=>"🅬",
 "1f16d"=>"🅭",
 "1f16e"=>"🅮",
 "1f16f"=>"🅯",
 "1f170"=>"🅰",
 "1f171"=>"🅱",
 "1f172"=>"🅲",
 "1f173"=>"🅳",
 "1f174"=>"🅴",
 "1f175"=>"🅵",
 "1f176"=>"🅶",
 "1f177"=>"🅷",
 "1f178"=>"🅸",
 "1f179"=>"🅹",
 "1f17a"=>"🅺",
 "1f17b"=>"🅻",
 "1f17c"=>"🅼",
 "1f17d"=>"🅽",
 "1f17e"=>"🅾",
 "1f17f"=>"🅿",
 "1f180"=>"🆀",
 "1f181"=>"🆁",
 "1f182"=>"🆂",
 "1f183"=>"🆃",
 "1f184"=>"🆄",
 "1f185"=>"🆅",
 "1f186"=>"🆆",
 "1f187"=>"🆇",
 "1f188"=>"🆈",
 "1f189"=>"🆉",
 "1f18a"=>"🆊",
 "1f18b"=>"🆋",
 "1f18c"=>"🆌",
 "1f18d"=>"🆍",
 "1f18e"=>"🆎",
 "1f18f"=>"🆏",
 "1f190"=>"🆐",
 "1f191"=>"🆑",
 "1f192"=>"🆒",
 "1f193"=>"🆓",
 "1f194"=>"🆔",
 "1f195"=>"🆕",
 "1f196"=>"🆖",
 "1f197"=>"🆗",
 "1f198"=>"🆘",
 "1f199"=>"🆙",
 "1f19a"=>"🆚",
 "1f19b"=>"🆛",
 "1f19c"=>"🆜",
 "1f19d"=>"🆝",
 "1f19e"=>"🆞",
 "1f19f"=>"🆟",
 "1f1a0"=>"🆠",
 "1f1a1"=>"🆡",
 "1f1a2"=>"🆢",
 "1f1a3"=>"🆣",
 "1f1a4"=>"🆤",
 "1f1a5"=>"🆥",
 "1f1a6"=>"🆦",
 "1f1a7"=>"🆧",
 "1f1a8"=>"🆨",
 "1f1a9"=>"🆩",
 "1f1aa"=>"🆪",
 "1f1ab"=>"🆫",
 "1f1ac"=>"🆬",
 "1f1ad"=>"🆭",
 "1f1ae"=>"\u{1F1AE}",
 "1f1af"=>"\u{1F1AF}",
 "1f1b0"=>"\u{1F1B0}",
 "1f1b1"=>"\u{1F1B1}",
 "1f1b2"=>"\u{1F1B2}",
 "1f1b3"=>"\u{1F1B3}",
 "1f1b4"=>"\u{1F1B4}",
 "1f1b5"=>"\u{1F1B5}",
 "1f1b6"=>"\u{1F1B6}",
 "1f1b7"=>"\u{1F1B7}",
 "1f1b8"=>"\u{1F1B8}",
 "1f1b9"=>"\u{1F1B9}",
 "1f1ba"=>"\u{1F1BA}",
 "1f1bb"=>"\u{1F1BB}",
 "1f1bc"=>"\u{1F1BC}",
 "1f1bd"=>"\u{1F1BD}",
 "1f1be"=>"\u{1F1BE}",
 "1f1bf"=>"\u{1F1BF}",
 "1f1c0"=>"\u{1F1C0}",
 "1f1c1"=>"\u{1F1C1}",
 "1f1c2"=>"\u{1F1C2}",
 "1f1c3"=>"\u{1F1C3}",
 "1f1c4"=>"\u{1F1C4}",
 "1f1c5"=>"\u{1F1C5}",
 "1f1c6"=>"\u{1F1C6}",
 "1f1c7"=>"\u{1F1C7}",
 "1f1c8"=>"\u{1F1C8}",
 "1f1c9"=>"\u{1F1C9}",
 "1f1ca"=>"\u{1F1CA}",
 "1f1cb"=>"\u{1F1CB}",
 "1f1cc"=>"\u{1F1CC}",
 "1f1cd"=>"\u{1F1CD}",
 "1f1ce"=>"\u{1F1CE}",
 "1f1cf"=>"\u{1F1CF}",
 "1f1d0"=>"\u{1F1D0}",
 "1f1d1"=>"\u{1F1D1}",
 "1f1d2"=>"\u{1F1D2}",
 "1f1d3"=>"\u{1F1D3}",
 "1f1d4"=>"\u{1F1D4}",
 "1f1d5"=>"\u{1F1D5}",
 "1f1d6"=>"\u{1F1D6}",
 "1f1d7"=>"\u{1F1D7}",
 "1f1d8"=>"\u{1F1D8}",
 "1f1d9"=>"\u{1F1D9}",
 "1f1da"=>"\u{1F1DA}",
 "1f1db"=>"\u{1F1DB}",
 "1f1dc"=>"\u{1F1DC}",
 "1f1dd"=>"\u{1F1DD}",
 "1f1de"=>"\u{1F1DE}",
 "1f1df"=>"\u{1F1DF}",
 "1f1e0"=>"\u{1F1E0}",
 "1f1e1"=>"\u{1F1E1}",
 "1f1e2"=>"\u{1F1E2}",
 "1f1e3"=>"\u{1F1E3}",
 "1f1e4"=>"\u{1F1E4}",
 "1f1e5"=>"\u{1F1E5}",
 "1f1e6"=>"🇦",
 "1f1e7"=>"🇧",
 "1f1e8"=>"🇨",
 "1f1e9"=>"🇩",
 "1f1ea"=>"🇪",
 "1f1eb"=>"🇫",
 "1f1ec"=>"🇬",
 "1f1ed"=>"🇭",
 "1f1ee"=>"🇮",
 "1f1ef"=>"🇯",
 "1f1f0"=>"🇰",
 "1f1f1"=>"🇱",
 "1f1f2"=>"🇲",
 "1f1f3"=>"🇳",
 "1f1f4"=>"🇴",
 "1f1f5"=>"🇵",
 "1f1f6"=>"🇶",
 "1f1f7"=>"🇷",
 "1f1f8"=>"🇸",
 "1f1f9"=>"🇹",
 "1f1fa"=>"🇺",
 "1f1fb"=>"🇻",
 "1f1fc"=>"🇼",
 "1f1fd"=>"🇽",
 "1f1fe"=>"🇾",
 "1f1ff"=>"🇿",
 "1f200"=>"🈀",
 "1f201"=>"🈁",
 "1f202"=>"🈂",
 "1f203"=>"\u{1F203}",
 "1f204"=>"\u{1F204}",
 "1f205"=>"\u{1F205}",
 "1f206"=>"\u{1F206}",
 "1f207"=>"\u{1F207}",
 "1f208"=>"\u{1F208}",
 "1f209"=>"\u{1F209}",
 "1f20a"=>"\u{1F20A}",
 "1f20b"=>"\u{1F20B}",
 "1f20c"=>"\u{1F20C}",
 "1f20d"=>"\u{1F20D}",
 "1f20e"=>"\u{1F20E}",
 "1f20f"=>"\u{1F20F}",
 "1f210"=>"🈐",
 "1f211"=>"🈑",
 "1f212"=>"🈒",
 "1f213"=>"🈓",
 "1f214"=>"🈔",
 "1f215"=>"🈕",
 "1f216"=>"🈖",
 "1f217"=>"🈗",
 "1f218"=>"🈘",
 "1f219"=>"🈙",
 "1f21a"=>"🈚",
 "1f21b"=>"🈛",
 "1f21c"=>"🈜",
 "1f21d"=>"🈝",
 "1f21e"=>"🈞",
 "1f21f"=>"🈟",
 "1f220"=>"🈠",
 "1f221"=>"🈡",
 "1f222"=>"🈢",
 "1f223"=>"🈣",
 "1f224"=>"🈤",
 "1f225"=>"🈥",
 "1f226"=>"🈦",
 "1f227"=>"🈧",
 "1f228"=>"🈨",
 "1f229"=>"🈩",
 "1f22a"=>"🈪",
 "1f22b"=>"🈫",
 "1f22c"=>"🈬",
 "1f22d"=>"🈭",
 "1f22e"=>"🈮",
 "1f22f"=>"🈯",
 "1f230"=>"🈰",
 "1f231"=>"🈱",
 "1f232"=>"🈲",
 "1f233"=>"🈳",
 "1f234"=>"🈴",
 "1f235"=>"🈵",
 "1f236"=>"🈶",
 "1f237"=>"🈷",
 "1f238"=>"🈸",
 "1f239"=>"🈹",
 "1f23a"=>"🈺",
 "1f23b"=>"🈻",
 "1f23c"=>"\u{1F23C}",
 "1f23d"=>"\u{1F23D}",
 "1f23e"=>"\u{1F23E}",
 "1f23f"=>"\u{1F23F}",
 "1f240"=>"🉀",
 "1f241"=>"🉁",
 "1f242"=>"🉂",
 "1f243"=>"🉃",
 "1f244"=>"🉄",
 "1f245"=>"🉅",
 "1f246"=>"🉆",
 "1f247"=>"🉇",
 "1f248"=>"🉈",
 "1f249"=>"\u{1F249}",
 "1f24a"=>"\u{1F24A}",
 "1f24b"=>"\u{1F24B}",
 "1f24c"=>"\u{1F24C}",
 "1f24d"=>"\u{1F24D}",
 "1f24e"=>"\u{1F24E}",
 "1f24f"=>"\u{1F24F}",
 "1f250"=>"🉐",
 "1f251"=>"🉑",
 "1f252"=>"\u{1F252}",
 "1f253"=>"\u{1F253}",
 "1f254"=>"\u{1F254}",
 "1f255"=>"\u{1F255}",
 "1f256"=>"\u{1F256}",
 "1f257"=>"\u{1F257}",
 "1f258"=>"\u{1F258}",
 "1f259"=>"\u{1F259}",
 "1f25a"=>"\u{1F25A}",
 "1f25b"=>"\u{1F25B}",
 "1f25c"=>"\u{1F25C}",
 "1f25d"=>"\u{1F25D}",
 "1f25e"=>"\u{1F25E}",
 "1f25f"=>"\u{1F25F}",
 "1f260"=>"🉠",
 "1f261"=>"🉡",
 "1f262"=>"🉢",
 "1f263"=>"🉣",
 "1f264"=>"🉤",
 "1f265"=>"🉥",
 "1f266"=>"\u{1F266}",
 "1f267"=>"\u{1F267}",
 "1f268"=>"\u{1F268}",
 "1f269"=>"\u{1F269}",
 "1f26a"=>"\u{1F26A}",
 "1f26b"=>"\u{1F26B}",
 "1f26c"=>"\u{1F26C}",
 "1f26d"=>"\u{1F26D}",
 "1f26e"=>"\u{1F26E}",
 "1f26f"=>"\u{1F26F}",
 "1f270"=>"\u{1F270}",
 "1f271"=>"\u{1F271}",
 "1f272"=>"\u{1F272}",
 "1f273"=>"\u{1F273}",
 "1f274"=>"\u{1F274}",
 "1f275"=>"\u{1F275}",
 "1f276"=>"\u{1F276}",
 "1f277"=>"\u{1F277}",
 "1f278"=>"\u{1F278}",
 "1f279"=>"\u{1F279}",
 "1f27a"=>"\u{1F27A}",
 "1f27b"=>"\u{1F27B}",
 "1f27c"=>"\u{1F27C}",
 "1f27d"=>"\u{1F27D}",
 "1f27e"=>"\u{1F27E}",
 "1f27f"=>"\u{1F27F}",
 "1f280"=>"\u{1F280}",
 "1f281"=>"\u{1F281}",
 "1f282"=>"\u{1F282}",
 "1f283"=>"\u{1F283}",
 "1f284"=>"\u{1F284}",
 "1f285"=>"\u{1F285}",
 "1f286"=>"\u{1F286}",
 "1f287"=>"\u{1F287}",
 "1f288"=>"\u{1F288}",
 "1f289"=>"\u{1F289}",
 "1f28a"=>"\u{1F28A}",
 "1f28b"=>"\u{1F28B}",
 "1f28c"=>"\u{1F28C}",
 "1f28d"=>"\u{1F28D}",
 "1f28e"=>"\u{1F28E}",
 "1f28f"=>"\u{1F28F}",
 "1f290"=>"\u{1F290}",
 "1f291"=>"\u{1F291}",
 "1f292"=>"\u{1F292}",
 "1f293"=>"\u{1F293}",
 "1f294"=>"\u{1F294}",
 "1f295"=>"\u{1F295}",
 "1f296"=>"\u{1F296}",
 "1f297"=>"\u{1F297}",
 "1f298"=>"\u{1F298}",
 "1f299"=>"\u{1F299}",
 "1f29a"=>"\u{1F29A}",
 "1f29b"=>"\u{1F29B}",
 "1f29c"=>"\u{1F29C}",
 "1f29d"=>"\u{1F29D}",
 "1f29e"=>"\u{1F29E}",
 "1f29f"=>"\u{1F29F}",
 "1f2a0"=>"\u{1F2A0}",
 "1f2a1"=>"\u{1F2A1}",
 "1f2a2"=>"\u{1F2A2}",
 "1f2a3"=>"\u{1F2A3}",
 "1f2a4"=>"\u{1F2A4}",
 "1f2a5"=>"\u{1F2A5}",
 "1f2a6"=>"\u{1F2A6}",
 "1f2a7"=>"\u{1F2A7}",
 "1f2a8"=>"\u{1F2A8}",
 "1f2a9"=>"\u{1F2A9}",
 "1f2aa"=>"\u{1F2AA}",
 "1f2ab"=>"\u{1F2AB}",
 "1f2ac"=>"\u{1F2AC}",
 "1f2ad"=>"\u{1F2AD}",
 "1f2ae"=>"\u{1F2AE}",
 "1f2af"=>"\u{1F2AF}",
 "1f2b0"=>"\u{1F2B0}",
 "1f2b1"=>"\u{1F2B1}",
 "1f2b2"=>"\u{1F2B2}",
 "1f2b3"=>"\u{1F2B3}",
 "1f2b4"=>"\u{1F2B4}",
 "1f2b5"=>"\u{1F2B5}",
 "1f2b6"=>"\u{1F2B6}",
 "1f2b7"=>"\u{1F2B7}",
 "1f2b8"=>"\u{1F2B8}",
 "1f2b9"=>"\u{1F2B9}",
 "1f2ba"=>"\u{1F2BA}",
 "1f2bb"=>"\u{1F2BB}",
 "1f2bc"=>"\u{1F2BC}",
 "1f2bd"=>"\u{1F2BD}",
 "1f2be"=>"\u{1F2BE}",
 "1f2bf"=>"\u{1F2BF}",
 "1f2c0"=>"\u{1F2C0}",
 "1f2c1"=>"\u{1F2C1}",
 "1f2c2"=>"\u{1F2C2}",
 "1f2c3"=>"\u{1F2C3}",
 "1f2c4"=>"\u{1F2C4}",
 "1f2c5"=>"\u{1F2C5}",
 "1f2c6"=>"\u{1F2C6}",
 "1f2c7"=>"\u{1F2C7}",
 "1f2c8"=>"\u{1F2C8}",
 "1f2c9"=>"\u{1F2C9}",
 "1f2ca"=>"\u{1F2CA}",
 "1f2cb"=>"\u{1F2CB}",
 "1f2cc"=>"\u{1F2CC}",
 "1f2cd"=>"\u{1F2CD}",
 "1f2ce"=>"\u{1F2CE}",
 "1f2cf"=>"\u{1F2CF}",
 "1f2d0"=>"\u{1F2D0}",
 "1f2d1"=>"\u{1F2D1}",
 "1f2d2"=>"\u{1F2D2}",
 "1f2d3"=>"\u{1F2D3}",
 "1f2d4"=>"\u{1F2D4}",
 "1f2d5"=>"\u{1F2D5}",
 "1f2d6"=>"\u{1F2D6}",
 "1f2d7"=>"\u{1F2D7}",
 "1f2d8"=>"\u{1F2D8}",
 "1f2d9"=>"\u{1F2D9}",
 "1f2da"=>"\u{1F2DA}",
 "1f2db"=>"\u{1F2DB}",
 "1f2dc"=>"\u{1F2DC}",
 "1f2dd"=>"\u{1F2DD}",
 "1f2de"=>"\u{1F2DE}",
 "1f2df"=>"\u{1F2DF}",
 "1f2e0"=>"\u{1F2E0}",
 "1f2e1"=>"\u{1F2E1}",
 "1f2e2"=>"\u{1F2E2}",
 "1f2e3"=>"\u{1F2E3}",
 "1f2e4"=>"\u{1F2E4}",
 "1f2e5"=>"\u{1F2E5}",
 "1f2e6"=>"\u{1F2E6}",
 "1f2e7"=>"\u{1F2E7}",
 "1f2e8"=>"\u{1F2E8}",
 "1f2e9"=>"\u{1F2E9}",
 "1f2ea"=>"\u{1F2EA}",
 "1f2eb"=>"\u{1F2EB}",
 "1f2ec"=>"\u{1F2EC}",
 "1f2ed"=>"\u{1F2ED}",
 "1f2ee"=>"\u{1F2EE}",
 "1f2ef"=>"\u{1F2EF}",
 "1f2f0"=>"\u{1F2F0}",
 "1f2f1"=>"\u{1F2F1}",
 "1f2f2"=>"\u{1F2F2}",
 "1f2f3"=>"\u{1F2F3}",
 "1f2f4"=>"\u{1F2F4}",
 "1f2f5"=>"\u{1F2F5}",
 "1f2f6"=>"\u{1F2F6}",
 "1f2f7"=>"\u{1F2F7}",
 "1f2f8"=>"\u{1F2F8}",
 "1f2f9"=>"\u{1F2F9}",
 "1f2fa"=>"\u{1F2FA}",
 "1f2fb"=>"\u{1F2FB}",
 "1f2fc"=>"\u{1F2FC}",
 "1f2fd"=>"\u{1F2FD}",
 "1f2fe"=>"\u{1F2FE}",
 "1f2ff"=>"\u{1F2FF}",
 "1f300"=>"🌀",
 "1f301"=>"🌁",
 "1f302"=>"🌂",
 "1f303"=>"🌃",
 "1f304"=>"🌄",
 "1f305"=>"🌅",
 "1f306"=>"🌆",
 "1f307"=>"🌇",
 "1f308"=>"🌈",
 "1f309"=>"🌉",
 "1f30a"=>"🌊",
 "1f30b"=>"🌋",
 "1f30c"=>"🌌",
 "1f30d"=>"🌍",
 "1f30e"=>"🌎",
 "1f30f"=>"🌏",
 "1f310"=>"🌐",
 "1f311"=>"🌑",
 "1f312"=>"🌒",
 "1f313"=>"🌓",
 "1f314"=>"🌔",
 "1f315"=>"🌕",
 "1f316"=>"🌖",
 "1f317"=>"🌗",
 "1f318"=>"🌘",
 "1f319"=>"🌙",
 "1f31a"=>"🌚",
 "1f31b"=>"🌛",
 "1f31c"=>"🌜",
 "1f31d"=>"🌝",
 "1f31e"=>"🌞",
 "1f31f"=>"🌟",
 "1f320"=>"🌠",
 "1f321"=>"🌡",
 "1f322"=>"🌢",
 "1f323"=>"🌣",
 "1f324"=>"🌤",
 "1f325"=>"🌥",
 "1f326"=>"🌦",
 "1f327"=>"🌧",
 "1f328"=>"🌨",
 "1f329"=>"🌩",
 "1f32a"=>"🌪",
 "1f32b"=>"🌫",
 "1f32c"=>"🌬",
 "1f32d"=>"🌭",
 "1f32e"=>"🌮",
 "1f32f"=>"🌯",
 "1f330"=>"🌰",
 "1f331"=>"🌱",
 "1f332"=>"🌲",
 "1f333"=>"🌳",
 "1f334"=>"🌴",
 "1f335"=>"🌵",
 "1f336"=>"🌶",
 "1f337"=>"🌷",
 "1f338"=>"🌸",
 "1f339"=>"🌹",
 "1f33a"=>"🌺",
 "1f33b"=>"🌻",
 "1f33c"=>"🌼",
 "1f33d"=>"🌽",
 "1f33e"=>"🌾",
 "1f33f"=>"🌿",
 "1f340"=>"🍀",
 "1f341"=>"🍁",
 "1f342"=>"🍂",
 "1f343"=>"🍃",
 "1f344"=>"🍄",
 "1f345"=>"🍅",
 "1f346"=>"🍆",
 "1f347"=>"🍇",
 "1f348"=>"🍈",
 "1f349"=>"🍉",
 "1f34a"=>"🍊",
 "1f34b"=>"🍋",
 "1f34c"=>"🍌",
 "1f34d"=>"🍍",
 "1f34e"=>"🍎",
 "1f34f"=>"🍏",
 "1f350"=>"🍐",
 "1f351"=>"🍑",
 "1f352"=>"🍒",
 "1f353"=>"🍓",
 "1f354"=>"🍔",
 "1f355"=>"🍕",
 "1f356"=>"🍖",
 "1f357"=>"🍗",
 "1f358"=>"🍘",
 "1f359"=>"🍙",
 "1f35a"=>"🍚",
 "1f35b"=>"🍛",
 "1f35c"=>"🍜",
 "1f35d"=>"🍝",
 "1f35e"=>"🍞",
 "1f35f"=>"🍟",
 "1f360"=>"🍠",
 "1f361"=>"🍡",
 "1f362"=>"🍢",
 "1f363"=>"🍣",
 "1f364"=>"🍤",
 "1f365"=>"🍥",
 "1f366"=>"🍦",
 "1f367"=>"🍧",
 "1f368"=>"🍨",
 "1f369"=>"🍩",
 "1f36a"=>"🍪",
 "1f36b"=>"🍫",
 "1f36c"=>"🍬",
 "1f36d"=>"🍭",
 "1f36e"=>"🍮",
 "1f36f"=>"🍯",
 "1f370"=>"🍰",
 "1f371"=>"🍱",
 "1f372"=>"🍲",
 "1f373"=>"🍳",
 "1f374"=>"🍴",
 "1f375"=>"🍵",
 "1f376"=>"🍶",
 "1f377"=>"🍷",
 "1f378"=>"🍸",
 "1f379"=>"🍹",
 "1f37a"=>"🍺",
 "1f37b"=>"🍻",
 "1f37c"=>"🍼",
 "1f37d"=>"🍽",
 "1f37e"=>"🍾",
 "1f37f"=>"🍿",
 "1f380"=>"🎀",
 "1f381"=>"🎁",
 "1f382"=>"🎂",
 "1f383"=>"🎃",
 "1f384"=>"🎄",
 "1f385"=>"🎅",
 "1f386"=>"🎆",
 "1f387"=>"🎇",
 "1f388"=>"🎈",
 "1f389"=>"🎉",
 "1f38a"=>"🎊",
 "1f38b"=>"🎋",
 "1f38c"=>"🎌",
 "1f38d"=>"🎍",
 "1f38e"=>"🎎",
 "1f38f"=>"🎏",
 "1f390"=>"🎐",
 "1f391"=>"🎑",
 "1f392"=>"🎒",
 "1f393"=>"🎓",
 "1f394"=>"🎔",
 "1f395"=>"🎕",
 "1f396"=>"🎖",
 "1f397"=>"🎗",
 "1f398"=>"🎘",
 "1f399"=>"🎙",
 "1f39a"=>"🎚",
 "1f39b"=>"🎛",
 "1f39c"=>"🎜",
 "1f39d"=>"🎝",
 "1f39e"=>"🎞",
 "1f39f"=>"🎟",
 "1f3a0"=>"🎠",
 "1f3a1"=>"🎡",
 "1f3a2"=>"🎢",
 "1f3a3"=>"🎣",
 "1f3a4"=>"🎤",
 "1f3a5"=>"🎥",
 "1f3a6"=>"🎦",
 "1f3a7"=>"🎧",
 "1f3a8"=>"🎨",
 "1f3a9"=>"🎩",
 "1f3aa"=>"🎪",
 "1f3ab"=>"🎫",
 "1f3ac"=>"🎬",
 "1f3ad"=>"🎭",
 "1f3ae"=>"🎮",
 "1f3af"=>"🎯",
 "1f3b0"=>"🎰",
 "1f3b1"=>"🎱",
 "1f3b2"=>"🎲",
 "1f3b3"=>"🎳",
 "1f3b4"=>"🎴",
 "1f3b5"=>"🎵",
 "1f3b6"=>"🎶",
 "1f3b7"=>"🎷",
 "1f3b8"=>"🎸",
 "1f3b9"=>"🎹",
 "1f3ba"=>"🎺",
 "1f3bb"=>"🎻",
 "1f3bc"=>"🎼",
 "1f3bd"=>"🎽",
 "1f3be"=>"🎾",
 "1f3bf"=>"🎿",
 "1f3c0"=>"🏀",
 "1f3c1"=>"🏁",
 "1f3c2"=>"🏂",
 "1f3c3"=>"🏃",
 "1f3c4"=>"🏄",
 "1f3c5"=>"🏅",
 "1f3c6"=>"🏆",
 "1f3c7"=>"🏇",
 "1f3c8"=>"🏈",
 "1f3c9"=>"🏉",
 "1f3ca"=>"🏊",
 "1f3cb"=>"🏋",
 "1f3cc"=>"🏌",
 "1f3cd"=>"🏍",
 "1f3ce"=>"🏎",
 "1f3cf"=>"🏏",
 "1f3d0"=>"🏐",
 "1f3d1"=>"🏑",
 "1f3d2"=>"🏒",
 "1f3d3"=>"🏓",
 "1f3d4"=>"🏔",
 "1f3d5"=>"🏕",
 "1f3d6"=>"🏖",
 "1f3d7"=>"🏗",
 "1f3d8"=>"🏘",
 "1f3d9"=>"🏙",
 "1f3da"=>"🏚",
 "1f3db"=>"🏛",
 "1f3dc"=>"🏜",
 "1f3dd"=>"🏝",
 "1f3de"=>"🏞",
 "1f3df"=>"🏟",
 "1f3e0"=>"🏠",
 "1f3e1"=>"🏡",
 "1f3e2"=>"🏢",
 "1f3e3"=>"🏣",
 "1f3e4"=>"🏤",
 "1f3e5"=>"🏥",
 "1f3e6"=>"🏦",
 "1f3e7"=>"🏧",
 "1f3e8"=>"🏨",
 "1f3e9"=>"🏩",
 "1f3ea"=>"🏪",
 "1f3eb"=>"🏫",
 "1f3ec"=>"🏬",
 "1f3ed"=>"🏭",
 "1f3ee"=>"🏮",
 "1f3ef"=>"🏯",
 "1f3f0"=>"🏰",
 "1f3f1"=>"🏱",
 "1f3f2"=>"🏲",
 "1f3f3"=>"🏳",
 "1f3f4"=>"🏴",
 "1f3f5"=>"🏵",
 "1f3f6"=>"🏶",
 "1f3f7"=>"🏷",
 "1f3f8"=>"🏸",
 "1f3f9"=>"🏹",
 "1f3fa"=>"🏺",
 "1f3fb"=>"🏻",
 "1f3fc"=>"🏼",
 "1f3fd"=>"🏽",
 "1f3fe"=>"🏾",
 "1f3ff"=>"🏿",
 "1f400"=>"🐀",
 "1f401"=>"🐁",
 "1f402"=>"🐂",
 "1f403"=>"🐃",
 "1f404"=>"🐄",
 "1f405"=>"🐅",
 "1f406"=>"🐆",
 "1f407"=>"🐇",
 "1f408"=>"🐈",
 "1f409"=>"🐉",
 "1f40a"=>"🐊",
 "1f40b"=>"🐋",
 "1f40c"=>"🐌",
 "1f40d"=>"🐍",
 "1f40e"=>"🐎",
 "1f40f"=>"🐏",
 "1f410"=>"🐐",
 "1f411"=>"🐑",
 "1f412"=>"🐒",
 "1f413"=>"🐓",
 "1f414"=>"🐔",
 "1f415"=>"🐕",
 "1f416"=>"🐖",
 "1f417"=>"🐗",
 "1f418"=>"🐘",
 "1f419"=>"🐙",
 "1f41a"=>"🐚",
 "1f41b"=>"🐛",
 "1f41c"=>"🐜",
 "1f41d"=>"🐝",
 "1f41e"=>"🐞",
 "1f41f"=>"🐟",
 "1f420"=>"🐠",
 "1f421"=>"🐡",
 "1f422"=>"🐢",
 "1f423"=>"🐣",
 "1f424"=>"🐤",
 "1f425"=>"🐥",
 "1f426"=>"🐦",
 "1f427"=>"🐧",
 "1f428"=>"🐨",
 "1f429"=>"🐩",
 "1f42a"=>"🐪",
 "1f42b"=>"🐫",
 "1f42c"=>"🐬",
 "1f42d"=>"🐭",
 "1f42e"=>"🐮",
 "1f42f"=>"🐯",
 "1f430"=>"🐰",
 "1f431"=>"🐱",
 "1f432"=>"🐲",
 "1f433"=>"🐳",
 "1f434"=>"🐴",
 "1f435"=>"🐵",
 "1f436"=>"🐶",
 "1f437"=>"🐷",
 "1f438"=>"🐸",
 "1f439"=>"🐹",
 "1f43a"=>"🐺",
 "1f43b"=>"🐻",
 "1f43c"=>"🐼",
 "1f43d"=>"🐽",
 "1f43e"=>"🐾",
 "1f43f"=>"🐿",
 "1f440"=>"👀",
 "1f441"=>"👁",
 "1f442"=>"👂",
 "1f443"=>"👃",
 "1f444"=>"👄",
 "1f445"=>"👅",
 "1f446"=>"👆",
 "1f447"=>"👇",
 "1f448"=>"👈",
 "1f449"=>"👉",
 "1f44a"=>"👊",
 "1f44b"=>"👋",
 "1f44c"=>"👌",
 "1f44d"=>"👍",
 "1f44e"=>"👎",
 "1f44f"=>"👏",
 "1f450"=>"👐",
 "1f451"=>"👑",
 "1f452"=>"👒",
 "1f453"=>"👓",
 "1f454"=>"👔",
 "1f455"=>"👕",
 "1f456"=>"👖",
 "1f457"=>"👗",
 "1f458"=>"👘",
 "1f459"=>"👙",
 "1f45a"=>"👚",
 "1f45b"=>"👛",
 "1f45c"=>"👜",
 "1f45d"=>"👝",
 "1f45e"=>"👞",
 "1f45f"=>"👟",
 "1f460"=>"👠",
 "1f461"=>"👡",
 "1f462"=>"👢",
 "1f463"=>"👣",
 "1f464"=>"👤",
 "1f465"=>"👥",
 "1f466"=>"👦",
 "1f467"=>"👧",
 "1f468"=>"👨",
 "1f469"=>"👩",
 "1f46a"=>"👪",
 "1f46b"=>"👫",
 "1f46c"=>"👬",
 "1f46d"=>"👭",
 "1f46e"=>"👮",
 "1f46f"=>"👯",
 "1f470"=>"👰",
 "1f471"=>"👱",
 "1f472"=>"👲",
 "1f473"=>"👳",
 "1f474"=>"👴",
 "1f475"=>"👵",
 "1f476"=>"👶",
 "1f477"=>"👷",
 "1f478"=>"👸",
 "1f479"=>"👹",
 "1f47a"=>"👺",
 "1f47b"=>"👻",
 "1f47c"=>"👼",
 "1f47d"=>"👽",
 "1f47e"=>"👾",
 "1f47f"=>"👿",
 "1f480"=>"💀",
 "1f481"=>"💁",
 "1f482"=>"💂",
 "1f483"=>"💃",
 "1f484"=>"💄",
 "1f485"=>"💅",
 "1f486"=>"💆",
 "1f487"=>"💇",
 "1f488"=>"💈",
 "1f489"=>"💉",
 "1f48a"=>"💊",
 "1f48b"=>"💋",
 "1f48c"=>"💌",
 "1f48d"=>"💍",
 "1f48e"=>"💎",
 "1f48f"=>"💏",
 "1f490"=>"💐",
 "1f491"=>"💑",
 "1f492"=>"💒",
 "1f493"=>"💓",
 "1f494"=>"💔",
 "1f495"=>"💕",
 "1f496"=>"💖",
 "1f497"=>"💗",
 "1f498"=>"💘",
 "1f499"=>"💙",
 "1f49a"=>"💚",
 "1f49b"=>"💛",
 "1f49c"=>"💜",
 "1f49d"=>"💝",
 "1f49e"=>"💞",
 "1f49f"=>"💟",
 "1f4a0"=>"💠",
 "1f4a1"=>"💡",
 "1f4a2"=>"💢",
 "1f4a3"=>"💣",
 "1f4a4"=>"💤",
 "1f4a5"=>"💥",
 "1f4a6"=>"💦",
 "1f4a7"=>"💧",
 "1f4a8"=>"💨",
 "1f4a9"=>"💩",
 "1f4aa"=>"💪",
 "1f4ab"=>"💫",
 "1f4ac"=>"💬",
 "1f4ad"=>"💭",
 "1f4ae"=>"💮",
 "1f4af"=>"💯",
 "1f4b0"=>"💰",
 "1f4b1"=>"💱",
 "1f4b2"=>"💲",
 "1f4b3"=>"💳",
 "1f4b4"=>"💴",
 "1f4b5"=>"💵",
 "1f4b6"=>"💶",
 "1f4b7"=>"💷",
 "1f4b8"=>"💸",
 "1f4b9"=>"💹",
 "1f4ba"=>"💺",
 "1f4bb"=>"💻",
 "1f4bc"=>"💼",
 "1f4bd"=>"💽",
 "1f4be"=>"💾",
 "1f4bf"=>"💿",
 "1f4c0"=>"📀",
 "1f4c1"=>"📁",
 "1f4c2"=>"📂",
 "1f4c3"=>"📃",
 "1f4c4"=>"📄",
 "1f4c5"=>"📅",
 "1f4c6"=>"📆",
 "1f4c7"=>"📇",
 "1f4c8"=>"📈",
 "1f4c9"=>"📉",
 "1f4ca"=>"📊",
 "1f4cb"=>"📋",
 "1f4cc"=>"📌",
 "1f4cd"=>"📍",
 "1f4ce"=>"📎",
 "1f4cf"=>"📏",
 "1f4d0"=>"📐",
 "1f4d1"=>"📑",
 "1f4d2"=>"📒",
 "1f4d3"=>"📓",
 "1f4d4"=>"📔",
 "1f4d5"=>"📕",
 "1f4d6"=>"📖",
 "1f4d7"=>"📗",
 "1f4d8"=>"📘",
 "1f4d9"=>"📙",
 "1f4da"=>"📚",
 "1f4db"=>"📛",
 "1f4dc"=>"📜",
 "1f4dd"=>"📝",
 "1f4de"=>"📞",
 "1f4df"=>"📟",
 "1f4e0"=>"📠",
 "1f4e1"=>"📡",
 "1f4e2"=>"📢",
 "1f4e3"=>"📣",
 "1f4e4"=>"📤",
 "1f4e5"=>"📥",
 "1f4e6"=>"📦",
 "1f4e7"=>"📧",
 "1f4e8"=>"📨",
 "1f4e9"=>"📩",
 "1f4ea"=>"📪",
 "1f4eb"=>"📫",
 "1f4ec"=>"📬",
 "1f4ed"=>"📭",
 "1f4ee"=>"📮",
 "1f4ef"=>"📯",
 "1f4f0"=>"📰",
 "1f4f1"=>"📱",
 "1f4f2"=>"📲",
 "1f4f3"=>"📳",
 "1f4f4"=>"📴",
 "1f4f5"=>"📵",
 "1f4f6"=>"📶",
 "1f4f7"=>"📷",
 "1f4f8"=>"📸",
 "1f4f9"=>"📹",
 "1f4fa"=>"📺",
 "1f4fb"=>"📻",
 "1f4fc"=>"📼",
 "1f4fd"=>"📽",
 "1f4fe"=>"📾",
 "1f4ff"=>"📿",
 "1f500"=>"🔀",
 "1f501"=>"🔁",
 "1f502"=>"🔂",
 "1f503"=>"🔃",
 "1f504"=>"🔄",
 "1f505"=>"🔅",
 "1f506"=>"🔆",
 "1f507"=>"🔇",
 "1f508"=>"🔈",
 "1f509"=>"🔉",
 "1f50a"=>"🔊",
 "1f50b"=>"🔋",
 "1f50c"=>"🔌",
 "1f50d"=>"🔍",
 "1f50e"=>"🔎",
 "1f50f"=>"🔏",
 "1f510"=>"🔐",
 "1f511"=>"🔑",
 "1f512"=>"🔒",
 "1f513"=>"🔓",
 "1f514"=>"🔔",
 "1f515"=>"🔕",
 "1f516"=>"🔖",
 "1f517"=>"🔗",
 "1f518"=>"🔘",
 "1f519"=>"🔙",
 "1f51a"=>"🔚",
 "1f51b"=>"🔛",
 "1f51c"=>"🔜",
 "1f51d"=>"🔝",
 "1f51e"=>"🔞",
 "1f51f"=>"🔟",
 "1f520"=>"🔠",
 "1f521"=>"🔡",
 "1f522"=>"🔢",
 "1f523"=>"🔣",
 "1f524"=>"🔤",
 "1f525"=>"🔥",
 "1f526"=>"🔦",
 "1f527"=>"🔧",
 "1f528"=>"🔨",
 "1f529"=>"🔩",
 "1f52a"=>"🔪",
 "1f52b"=>"🔫",
 "1f52c"=>"🔬",
 "1f52d"=>"🔭",
 "1f52e"=>"🔮",
 "1f52f"=>"🔯",
 "1f530"=>"🔰",
 "1f531"=>"🔱",
 "1f532"=>"🔲",
 "1f533"=>"🔳",
 "1f534"=>"🔴",
 "1f535"=>"🔵",
 "1f536"=>"🔶",
 "1f537"=>"🔷",
 "1f538"=>"🔸",
 "1f539"=>"🔹",
 "1f53a"=>"🔺",
 "1f53b"=>"🔻",
 "1f53c"=>"🔼",
 "1f53d"=>"🔽",
 "1f53e"=>"🔾",
 "1f53f"=>"🔿",
 "1f540"=>"🕀",
 "1f541"=>"🕁",
 "1f542"=>"🕂",
 "1f543"=>"🕃",
 "1f544"=>"🕄",
 "1f545"=>"🕅",
 "1f546"=>"🕆",
 "1f547"=>"🕇",
 "1f548"=>"🕈",
 "1f549"=>"🕉",
 "1f54a"=>"🕊",
 "1f54b"=>"🕋",
 "1f54c"=>"🕌",
 "1f54d"=>"🕍",
 "1f54e"=>"🕎",
 "1f54f"=>"🕏",
 "1f550"=>"🕐",
 "1f551"=>"🕑",
 "1f552"=>"🕒",
 "1f553"=>"🕓",
 "1f554"=>"🕔",
 "1f555"=>"🕕",
 "1f556"=>"🕖",
 "1f557"=>"🕗",
 "1f558"=>"🕘",
 "1f559"=>"🕙",
 "1f55a"=>"🕚",
 "1f55b"=>"🕛",
 "1f55c"=>"🕜",
 "1f55d"=>"🕝",
 "1f55e"=>"🕞",
 "1f55f"=>"🕟",
 "1f560"=>"🕠",
 "1f561"=>"🕡",
 "1f562"=>"🕢",
 "1f563"=>"🕣",
 "1f564"=>"🕤",
 "1f565"=>"🕥",
 "1f566"=>"🕦",
 "1f567"=>"🕧",
 "1f568"=>"🕨",
 "1f569"=>"🕩",
 "1f56a"=>"🕪",
 "1f56b"=>"🕫",
 "1f56c"=>"🕬",
 "1f56d"=>"🕭",
 "1f56e"=>"🕮",
 "1f56f"=>"🕯",
 "1f570"=>"🕰",
 "1f571"=>"🕱",
 "1f572"=>"🕲",
 "1f573"=>"🕳",
 "1f574"=>"🕴",
 "1f575"=>"🕵",
 "1f576"=>"🕶",
 "1f577"=>"🕷",
 "1f578"=>"🕸",
 "1f579"=>"🕹",
 "1f57a"=>"🕺",
 "1f57b"=>"🕻",
 "1f57c"=>"🕼",
 "1f57d"=>"🕽",
 "1f57e"=>"🕾",
 "1f57f"=>"🕿",
 "1f580"=>"🖀",
 "1f581"=>"🖁",
 "1f582"=>"🖂",
 "1f583"=>"🖃",
 "1f584"=>"🖄",
 "1f585"=>"🖅",
 "1f586"=>"🖆",
 "1f587"=>"🖇",
 "1f588"=>"🖈",
 "1f589"=>"🖉",
 "1f58a"=>"🖊",
 "1f58b"=>"🖋",
 "1f58c"=>"🖌",
 "1f58d"=>"🖍",
 "1f58e"=>"🖎",
 "1f58f"=>"🖏",
 "1f590"=>"🖐",
 "1f591"=>"🖑",
 "1f592"=>"🖒",
 "1f593"=>"🖓",
 "1f594"=>"🖔",
 "1f595"=>"🖕",
 "1f596"=>"🖖",
 "1f597"=>"🖗",
 "1f598"=>"🖘",
 "1f599"=>"🖙",
 "1f59a"=>"🖚",
 "1f59b"=>"🖛",
 "1f59c"=>"🖜",
 "1f59d"=>"🖝",
 "1f59e"=>"🖞",
 "1f59f"=>"🖟",
 "1f5a0"=>"🖠",
 "1f5a1"=>"🖡",
 "1f5a2"=>"🖢",
 "1f5a3"=>"🖣",
 "1f5a4"=>"🖤",
 "1f5a5"=>"🖥",
 "1f5a6"=>"🖦",
 "1f5a7"=>"🖧",
 "1f5a8"=>"🖨",
 "1f5a9"=>"🖩",
 "1f5aa"=>"🖪",
 "1f5ab"=>"🖫",
 "1f5ac"=>"🖬",
 "1f5ad"=>"🖭",
 "1f5ae"=>"🖮",
 "1f5af"=>"🖯",
 "1f5b0"=>"🖰",
 "1f5b1"=>"🖱",
 "1f5b2"=>"🖲",
 "1f5b3"=>"🖳",
 "1f5b4"=>"🖴",
 "1f5b5"=>"🖵",
 "1f5b6"=>"🖶",
 "1f5b7"=>"🖷",
 "1f5b8"=>"🖸",
 "1f5b9"=>"🖹",
 "1f5ba"=>"🖺",
 "1f5bb"=>"🖻",
 "1f5bc"=>"🖼",
 "1f5bd"=>"🖽",
 "1f5be"=>"🖾",
 "1f5bf"=>"🖿",
 "1f5c0"=>"🗀",
 "1f5c1"=>"🗁",
 "1f5c2"=>"🗂",
 "1f5c3"=>"🗃",
 "1f5c4"=>"🗄",
 "1f5c5"=>"🗅",
 "1f5c6"=>"🗆",
 "1f5c7"=>"🗇",
 "1f5c8"=>"🗈",
 "1f5c9"=>"🗉",
 "1f5ca"=>"🗊",
 "1f5cb"=>"🗋",
 "1f5cc"=>"🗌",
 "1f5cd"=>"🗍",
 "1f5ce"=>"🗎",
 "1f5cf"=>"🗏",
 "1f5d0"=>"🗐",
 "1f5d1"=>"🗑",
 "1f5d2"=>"🗒",
 "1f5d3"=>"🗓",
 "1f5d4"=>"🗔",
 "1f5d5"=>"🗕",
 "1f5d6"=>"🗖",
 "1f5d7"=>"🗗",
 "1f5d8"=>"🗘",
 "1f5d9"=>"🗙",
 "1f5da"=>"🗚",
 "1f5db"=>"🗛",
 "1f5dc"=>"🗜",
 "1f5dd"=>"🗝",
 "1f5de"=>"🗞",
 "1f5df"=>"🗟",
 "1f5e0"=>"🗠",
 "1f5e1"=>"🗡",
 "1f5e2"=>"🗢",
 "1f5e3"=>"🗣",
 "1f5e4"=>"🗤",
 "1f5e5"=>"🗥",
 "1f5e6"=>"🗦",
 "1f5e7"=>"🗧",
 "1f5e8"=>"🗨",
 "1f5e9"=>"🗩",
 "1f5ea"=>"🗪",
 "1f5eb"=>"🗫",
 "1f5ec"=>"🗬",
 "1f5ed"=>"🗭",
 "1f5ee"=>"🗮",
 "1f5ef"=>"🗯",
 "1f5f0"=>"🗰",
 "1f5f1"=>"🗱",
 "1f5f2"=>"🗲",
 "1f5f3"=>"🗳",
 "1f5f4"=>"🗴",
 "1f5f5"=>"🗵",
 "1f5f6"=>"🗶",
 "1f5f7"=>"🗷",
 "1f5f8"=>"🗸",
 "1f5f9"=>"🗹",
 "1f5fa"=>"🗺",
 "1f5fb"=>"🗻",
 "1f5fc"=>"🗼",
 "1f5fd"=>"🗽",
 "1f5fe"=>"🗾",
 "1f5ff"=>"🗿",
 "1f600"=>"😀",
 "1f601"=>"😁",
 "1f602"=>"😂",
 "1f603"=>"😃",
 "1f604"=>"😄",
 "1f605"=>"😅",
 "1f606"=>"😆",
 "1f607"=>"😇",
 "1f608"=>"😈",
 "1f609"=>"😉",
 "1f60a"=>"😊",
 "1f60b"=>"😋",
 "1f60c"=>"😌",
 "1f60d"=>"😍",
 "1f60e"=>"😎",
 "1f60f"=>"😏",
 "1f610"=>"😐",
 "1f611"=>"😑",
 "1f612"=>"😒",
 "1f613"=>"😓",
 "1f614"=>"😔",
 "1f615"=>"😕",
 "1f616"=>"😖",
 "1f617"=>"😗",
 "1f618"=>"😘",
 "1f619"=>"😙",
 "1f61a"=>"😚",
 "1f61b"=>"😛",
 "1f61c"=>"😜",
 "1f61d"=>"😝",
 "1f61e"=>"😞",
 "1f61f"=>"😟",
 "1f620"=>"😠",
 "1f621"=>"😡",
 "1f622"=>"😢",
 "1f623"=>"😣",
 "1f624"=>"😤",
 "1f625"=>"😥",
 "1f626"=>"😦",
 "1f627"=>"😧",
 "1f628"=>"😨",
 "1f629"=>"😩",
 "1f62a"=>"😪",
 "1f62b"=>"😫",
 "1f62c"=>"😬",
 "1f62d"=>"😭",
 "1f62e"=>"😮",
 "1f62f"=>"😯",
 "1f630"=>"😰",
 "1f631"=>"😱",
 "1f632"=>"😲",
 "1f633"=>"😳",
 "1f634"=>"😴",
 "1f635"=>"😵",
 "1f636"=>"😶",
 "1f637"=>"😷",
 "1f638"=>"😸",
 "1f639"=>"😹",
 "1f63a"=>"😺",
 "1f63b"=>"😻",
 "1f63c"=>"😼",
 "1f63d"=>"😽",
 "1f63e"=>"😾",
 "1f63f"=>"😿",
 "1f640"=>"🙀",
 "1f641"=>"🙁",
 "1f642"=>"🙂",
 "1f643"=>"🙃",
 "1f644"=>"🙄",
 "1f645"=>"🙅",
 "1f646"=>"🙆",
 "1f647"=>"🙇",
 "1f648"=>"🙈",
 "1f649"=>"🙉",
 "1f64a"=>"🙊",
 "1f64b"=>"🙋",
 "1f64c"=>"🙌",
 "1f64d"=>"🙍",
 "1f64e"=>"🙎",
 "1f64f"=>"🙏",
 "1f680"=>"🚀",
 "1f681"=>"🚁",
 "1f682"=>"🚂",
 "1f683"=>"🚃",
 "1f684"=>"🚄",
 "1f685"=>"🚅",
 "1f686"=>"🚆",
 "1f687"=>"🚇",
 "1f688"=>"🚈",
 "1f689"=>"🚉",
 "1f68a"=>"🚊",
 "1f68b"=>"🚋",
 "1f68c"=>"🚌",
 "1f68d"=>"🚍",
 "1f68e"=>"🚎",
 "1f68f"=>"🚏",
 "1f690"=>"🚐",
 "1f691"=>"🚑",
 "1f692"=>"🚒",
 "1f693"=>"🚓",
 "1f694"=>"🚔",
 "1f695"=>"🚕",
 "1f696"=>"🚖",
 "1f697"=>"🚗",
 "1f698"=>"🚘",
 "1f699"=>"🚙",
 "1f69a"=>"🚚",
 "1f69b"=>"🚛",
 "1f69c"=>"🚜",
 "1f69d"=>"🚝",
 "1f69e"=>"🚞",
 "1f69f"=>"🚟",
 "1f6a0"=>"🚠",
 "1f6a1"=>"🚡",
 "1f6a2"=>"🚢",
 "1f6a3"=>"🚣",
 "1f6a4"=>"🚤",
 "1f6a5"=>"🚥",
 "1f6a6"=>"🚦",
 "1f6a7"=>"🚧",
 "1f6a8"=>"🚨",
 "1f6a9"=>"🚩",
 "1f6aa"=>"🚪",
 "1f6ab"=>"🚫",
 "1f6ac"=>"🚬",
 "1f6ad"=>"🚭",
 "1f6ae"=>"🚮",
 "1f6af"=>"🚯",
 "1f6b0"=>"🚰",
 "1f6b1"=>"🚱",
 "1f6b2"=>"🚲",
 "1f6b3"=>"🚳",
 "1f6b4"=>"🚴",
 "1f6b5"=>"🚵",
 "1f6b6"=>"🚶",
 "1f6b7"=>"🚷",
 "1f6b8"=>"🚸",
 "1f6b9"=>"🚹",
 "1f6ba"=>"🚺",
 "1f6bb"=>"🚻",
 "1f6bc"=>"🚼",
 "1f6bd"=>"🚽",
 "1f6be"=>"🚾",
 "1f6bf"=>"🚿",
 "1f6c0"=>"🛀",
 "1f6c1"=>"🛁",
 "1f6c2"=>"🛂",
 "1f6c3"=>"🛃",
 "1f6c4"=>"🛄",
 "1f6c5"=>"🛅",
 "1f6c6"=>"🛆",
 "1f6c7"=>"🛇",
 "1f6c8"=>"🛈",
 "1f6c9"=>"🛉",
 "1f6ca"=>"🛊",
 "1f6cb"=>"🛋",
 "1f6cc"=>"🛌",
 "1f6cd"=>"🛍",
 "1f6ce"=>"🛎",
 "1f6cf"=>"🛏",
 "1f6d0"=>"🛐",
 "1f6d1"=>"🛑",
 "1f6d2"=>"🛒",
 "1f6d3"=>"🛓",
 "1f6d4"=>"🛔",
 "1f6d5"=>"🛕",
 "1f6d6"=>"🛖",
 "1f6d7"=>"🛗",
 "1f6d8"=>"\u{1F6D8}",
 "1f6d9"=>"\u{1F6D9}",
 "1f6da"=>"\u{1F6DA}",
 "1f6db"=>"\u{1F6DB}",
 "1f6dc"=>"\u{1F6DC}",
 "1f6dd"=>"\u{1F6DD}",
 "1f6de"=>"\u{1F6DE}",
 "1f6df"=>"\u{1F6DF}",
 "1f6e0"=>"🛠",
 "1f6e1"=>"🛡",
 "1f6e2"=>"🛢",
 "1f6e3"=>"🛣",
 "1f6e4"=>"🛤",
 "1f6e5"=>"🛥",
 "1f6e6"=>"🛦",
 "1f6e7"=>"🛧",
 "1f6e8"=>"🛨",
 "1f6e9"=>"🛩",
 "1f6ea"=>"🛪",
 "1f6eb"=>"🛫",
 "1f6ec"=>"🛬",
 "1f6ed"=>"\u{1F6ED}",
 "1f6ee"=>"\u{1F6EE}",
 "1f6ef"=>"\u{1F6EF}",
 "1f6f0"=>"🛰",
 "1f6f1"=>"🛱",
 "1f6f2"=>"🛲",
 "1f6f3"=>"🛳",
 "1f6f4"=>"🛴",
 "1f6f5"=>"🛵",
 "1f6f6"=>"🛶",
 "1f6f7"=>"🛷",
 "1f6f8"=>"🛸",
 "1f6f9"=>"🛹",
 "1f6fa"=>"🛺",
 "1f6fb"=>"🛻",
 "1f6fc"=>"🛼",
 "1f6fd"=>"\u{1F6FD}",
 "1f6fe"=>"\u{1F6FE}",
 "1f6ff"=>"\u{1F6FF}",
 "1f910"=>"🤐",
 "1f911"=>"🤑",
 "1f912"=>"🤒",
 "1f913"=>"🤓",
 "1f914"=>"🤔",
 "1f915"=>"🤕",
 "1f916"=>"🤖",
 "1f917"=>"🤗",
 "1f918"=>"🤘",
 "1f919"=>"🤙",
 "1f91a"=>"🤚",
 "1f91b"=>"🤛",
 "1f91c"=>"🤜",
 "1f91d"=>"🤝",
 "1f91e"=>"🤞",
 "1f91f"=>"🤟",
 "1f920"=>"🤠",
 "1f921"=>"🤡",
 "1f922"=>"🤢",
 "1f923"=>"🤣",
 "1f924"=>"🤤",
 "1f925"=>"🤥",
 "1f926"=>"🤦",
 "1f927"=>"🤧",
 "1f928"=>"🤨",
 "1f929"=>"🤩",
 "1f92a"=>"🤪",
 "1f92b"=>"🤫",
 "1f92c"=>"🤬",
 "1f92d"=>"🤭",
 "1f92e"=>"🤮",
 "1f92f"=>"🤯",
 "1f930"=>"🤰",
 "1f931"=>"🤱",
 "1f932"=>"🤲",
 "1f933"=>"🤳",
 "1f934"=>"🤴",
 "1f935"=>"🤵",
 "1f936"=>"🤶",
 "1f937"=>"🤷",
 "1f938"=>"🤸",
 "1f939"=>"🤹",
 "1f93a"=>"🤺",
 "1f93b"=>"🤻",
 "1f93c"=>"🤼",
 "1f93d"=>"🤽",
 "1f93e"=>"🤾",
 "1f93f"=>"🤿",
 "1f940"=>"🥀",
 "1f941"=>"🥁",
 "1f942"=>"🥂",
 "1f943"=>"🥃",
 "1f944"=>"🥄",
 "1f945"=>"🥅",
 "1f946"=>"🥆",
 "1f947"=>"🥇",
 "1f948"=>"🥈",
 "1f949"=>"🥉",
 "1f94a"=>"🥊",
 "1f94b"=>"🥋",
 "1f94c"=>"🥌",
 "1f94d"=>"🥍",
 "1f94e"=>"🥎",
 "1f94f"=>"🥏",
 "1f950"=>"🥐",
 "1f951"=>"🥑",
 "1f952"=>"🥒",
 "1f953"=>"🥓",
 "1f954"=>"🥔",
 "1f955"=>"🥕",
 "1f956"=>"🥖",
 "1f957"=>"🥗",
 "1f958"=>"🥘",
 "1f959"=>"🥙",
 "1f95a"=>"🥚",
 "1f95b"=>"🥛",
 "1f95c"=>"🥜",
 "1f95d"=>"🥝",
 "1f95e"=>"🥞",
 "1f95f"=>"🥟",
 "1f960"=>"🥠",
 "1f961"=>"🥡",
 "1f962"=>"🥢",
 "1f963"=>"🥣",
 "1f964"=>"🥤",
 "1f965"=>"🥥",
 "1f966"=>"🥦",
 "1f967"=>"🥧",
 "1f968"=>"🥨",
 "1f969"=>"🥩",
 "1f96a"=>"🥪",
 "1f96b"=>"🥫",
 "1f980"=>"🦀",
 "1f981"=>"🦁",
 "1f982"=>"🦂",
 "1f983"=>"🦃",
 "1f984"=>"🦄",
 "1f985"=>"🦅",
 "1f986"=>"🦆",
 "1f987"=>"🦇",
 "1f988"=>"🦈",
 "1f989"=>"🦉",
 "1f98a"=>"🦊",
 "1f98b"=>"🦋",
 "1f98c"=>"🦌",
 "1f98d"=>"🦍",
 "1f98e"=>"🦎",
 "1f98f"=>"🦏",
 "1f990"=>"🦐",
 "1f991"=>"🦑",
 "1f992"=>"🦒",
 "1f993"=>"🦓",
 "1f994"=>"🦔",
 "1f995"=>"🦕",
 "1f996"=>"🦖",
 "1f997"=>"🦗",
 "1f998"=>"🦘",
 "1f999"=>"🦙",
 "1f99a"=>"🦚",
 "1f99b"=>"🦛",
 "1f99c"=>"🦜",
 "1f99d"=>"🦝",
 "1f99e"=>"🦞",
 "1f99f"=>"🦟",
 "1f9a0"=>"🦠",
 "1f9a1"=>"🦡",
 "1f9a2"=>"🦢",
 "1f9a3"=>"🦣",
 "1f9a4"=>"🦤",
 "1f9a5"=>"🦥",
 "1f9a6"=>"🦦",
 "1f9a7"=>"🦧",
 "1f9a8"=>"🦨",
 "1f9a9"=>"🦩",
 "1f9aa"=>"🦪",
 "1f9ab"=>"🦫",
 "1f9ac"=>"🦬",
 "1f9ad"=>"🦭",
 "1f9ae"=>"🦮",
 "1f9af"=>"🦯",
 "1f9b0"=>"🦰",
 "1f9b1"=>"🦱",
 "1f9b2"=>"🦲",
 "1f9b3"=>"🦳",
 "1f9b4"=>"🦴",
 "1f9b5"=>"🦵",
 "1f9b6"=>"🦶",
 "1f9b7"=>"🦷",
 "1f9b8"=>"🦸",
 "1f9b9"=>"🦹",
 "1f9ba"=>"🦺",
 "1f9bb"=>"🦻",
 "1f9bc"=>"🦼",
 "1f9bd"=>"🦽",
 "1f9be"=>"🦾",
 "1f9bf"=>"🦿",
 "1f9c0"=>"🧀",
 "1f9c1"=>"🧁",
 "1f9c2"=>"🧂",
 "1f9c3"=>"🧃",
 "1f9c4"=>"🧄",
 "1f9c5"=>"🧅",
 "1f9c6"=>"🧆",
 "1f9c7"=>"🧇",
 "1f9c8"=>"🧈",
 "1f9c9"=>"🧉",
 "1f9ca"=>"🧊",
 "1f9cb"=>"🧋",
 "1f9cc"=>"\u{1F9CC}",
 "1f9cd"=>"🧍",
 "1f9ce"=>"🧎",
 "1f9cf"=>"🧏",
 "1f9d0"=>"🧐",
 "1f9d1"=>"🧑",
 "1f9d2"=>"🧒",
 "1f9d3"=>"🧓",
 "1f9d4"=>"🧔",
 "1f9d5"=>"🧕",
 "1f9d6"=>"🧖",
 "1f9d7"=>"🧗",
 "1f9d8"=>"🧘",
 "1f9d9"=>"🧙",
 "1f9da"=>"🧚",
 "1f9db"=>"🧛",
 "1f9dc"=>"🧜",
 "1f9dd"=>"🧝",
 "1f9de"=>"🧞",
 "1f9df"=>"🧟",
 "1f9e0"=>"🧠"}

</details>

It looks like it filters out fake space characters, like the `2800` used in the post that sparked this.

jgaskins commented:
I finally got Rubocop to accept my changes but it looks like it also took the liberty of "fixing" a bunch of other things in this file. These were not intentional changes on my part.
